{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import math\n",
    "import os\n",
    "from sklearn.cluster import KMeans"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hide_input": true
   },
   "outputs": [],
   "source": [
    "def lunghezza_max (d):  \n",
    "    lunghezza_max=0\n",
    "    for lista in d.values():\n",
    "        if len(lista)> lunghezza_max:\n",
    "            lunghezza_max= len(lista)\n",
    "    return lunghezza_max\n",
    "\n",
    "\n",
    "def completa_Nan (d):  \n",
    "    for lista in d.values():\n",
    "        for i in range(0, lunghezza_max(d) - len(lista)):\n",
    "            lista.append(np.NAN)\n",
    "            \n",
    "def trova_chiave_da(dizionario,simbolo):  \n",
    "    for key in dizionario:\n",
    "        if dizionario[key]== simbolo:\n",
    "            return key\n",
    "        \n",
    "        \n",
    "def campione_random (diz_vuoto, n_campione):\n",
    "    lista_sample_n= np.random.choice(lista_idgeni, size=n_campione, replace=False)\n",
    "    for idgene in lista_sample_n:\n",
    "        diz_vuoto[idgene]= annotazioni_pulito_id[idgene]\n",
    "    return diz_vuoto\n",
    "\n",
    "\n",
    "def count_campione_geni (dizio_geni, dizio_count):\n",
    "    for lista_funzioni in dizio_geni.values():\n",
    "        for funzione in lista_funzioni:\n",
    "            if funzione not in dizio_count:\n",
    "                dizio_count[funzione]=1\n",
    "            else:\n",
    "                dizio_count[funzione]+=1\n",
    "    return dizio_count\n",
    "\n",
    "\n",
    "def tabella_da_diz (diz_count_annotazioni):\n",
    "    series_annotaz_campione= pd.Series(diz_count_annotazioni)  \n",
    "    valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "    tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "    tab_funzioni_random.columns=['in sample 1']\n",
    "    return tab_funzioni_random\n",
    "\n",
    "\n",
    "def colonna_random (diz_vuoto, dizio_count, n_campione):\n",
    "    lista_sample_n= np.random.choice(lista_idgeni, size=n_campione, replace=False)\n",
    "    for idgene in lista_sample_n:\n",
    "        diz_vuoto[idgene]= annotazioni_pulito_id[idgene]\n",
    "    for lista_funzioni in diz_vuoto.values():\n",
    "        for funzione in lista_funzioni:\n",
    "            if funzione not in dizio_count:\n",
    "                dizio_count[funzione]=1\n",
    "            else:\n",
    "                dizio_count[funzione]+=1\n",
    "    series_annotaz_campione= pd.Series(dizio_count)  \n",
    "    valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "    tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "    tab_funzioni_random.columns= ['rand sample']\n",
    "    return tab_funzioni_random\n",
    "\n",
    "\n",
    "\n",
    "def aggiungi_colonna(dataframe, nuova_colonna):\n",
    "    dataframe= dataframe.join(nuova_colonna)  #.sort_values(by= 'n occorrenze funzione in natdisease', ascending=False)\n",
    "    return dataframe\n",
    "\n",
    "\n",
    "def diz_annotazioni_da_lista (diz_vuoto, lista_di_geni, dizio_count, titolo_colonna):\n",
    "    for idgene in lista_di_geni:\n",
    "        if idgene in annotazioni_pulito_id:\n",
    "            diz_vuoto[idgene]= annotazioni_pulito_id[idgene]\n",
    "        else:\n",
    "            try:\n",
    "                if idgene not in geni_sconosciuti:\n",
    "                    geni_sconosciuti.append(int(idgene))\n",
    "            except:\n",
    "                pass\n",
    "                \n",
    "    for lista_funzioni in diz_vuoto.values():\n",
    "        for funzione in lista_funzioni:\n",
    "            if funzione not in dizio_count:\n",
    "                dizio_count[funzione]=1\n",
    "            else:\n",
    "                dizio_count[funzione]+=1\n",
    "    series_annotaz_campione= pd.Series(dizio_count)  \n",
    "    valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "    tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "    tab_funzioni_random.columns=[titolo_colonna]\n",
    "    return tab_funzioni_random\n",
    "\n",
    "def funzioni_da_IDgene(geneid): \n",
    "    simbolo= idgene_simbolo[geneid]\n",
    "    funzioni= annotazioni_pulito[simbolo]\n",
    "    return simbolo, funzioni\n",
    "\n",
    "\n",
    "def colonna_random_mod (diz_vuoto, dizio_count, n_campione, titolo_colonna):\n",
    "    lista_sample_n= np.random.choice(filtro_seeds_carboidrati, size=n_campione, replace=False)\n",
    "    for idgene in lista_sample_n:\n",
    "        if idgene in annotazioni_pulito_id.keys():\n",
    "            diz_vuoto[idgene]= annotazioni_pulito_id[idgene]\n",
    "    for lista_funzioni in diz_vuoto.values():\n",
    "        for funzione in lista_funzioni:\n",
    "            if funzione not in dizio_count:\n",
    "                dizio_count[funzione]=1\n",
    "            else:\n",
    "                dizio_count[funzione]+=1\n",
    "    series_annotaz_campione= pd.Series(dizio_count)  \n",
    "    valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "    tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "    tab_funzioni_random.columns= [titolo_colonna]\n",
    "    return tab_funzioni_random\n",
    "\n",
    "def tabella_random_id (diz_vuoto, dizio_count, size_campione, numero_campioni, lista_geni_filtrata):\n",
    "    for i in range(1,numero_campioni+1):  \n",
    "        diz_vuoto={}\n",
    "        dizio_count={}\n",
    "        lista_sample_n= np.random.choice(lista_geni_filtrata, size= size_campione, replace=False)  \n",
    "        for idgene in lista_sample_n:\n",
    "            if idgene in annotazioni_pulito_id.keys():\n",
    "                diz_vuoto[idgene]= annotazioni_pulito_id[idgene]\n",
    "        for lista_funzioni in diz_vuoto.values():\n",
    "            for funzione in lista_funzioni:\n",
    "                if funzione not in dizio_count:\n",
    "                    dizio_count[funzione]=1\n",
    "                else:\n",
    "                    dizio_count[funzione]+=1\n",
    "        series_annotaz_campione= pd.Series(dizio_count)  \n",
    "        valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "        if i==1:\n",
    "            col_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            col_funzioni_random.columns= ['random sample 1']\n",
    "        elif i>1:\n",
    "            tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            tab_funzioni_random.columns= ['random sample ' + str(i)]\n",
    "            col_funzioni_random= col_funzioni_random.join(tab_funzioni_random)\n",
    "    return col_funzioni_random\n",
    "\n",
    "def media_tab_random (tab_campioni_random, num_campioni_usati):\n",
    "    media_campioni_random= pd.DataFrame(tab_campioni_random).mean(axis=1)\n",
    "    media_campioni_random.sort_values(ascending=False,inplace=True)\n",
    "    media_campioni_random = pd.DataFrame(media_campioni_random)\n",
    "    media_campioni_random.columns= ['media count annotazioni su ' + str(num_campioni_usati) +' campioni random']\n",
    "    return media_campioni_random\n",
    "\n",
    "def lista_n_annotazioni ( tab_media_random, tab_annot_malattia, n1, n2): \n",
    "    df= tab_annot_malattia.join(tab_media_random)\n",
    "    df.fillna(0,inplace=True)\n",
    "    top_10_malattia= list(df.iloc[0:n1].index)\n",
    "    df2= tab_media_random.join(tab_annot_malattia)\n",
    "    df2.fillna(0,inplace=True)\n",
    "    top_10_random= list(df2.iloc[0:n2].index)\n",
    "    \n",
    "    return top_10_malattia + top_10_random\n",
    "\n",
    "\n",
    "def geni_con_top_annotazioni (nome_lista_geni, lista_top_annotazioni):\n",
    "    nome_lista_geni= []\n",
    "    for gene in annotazioni_pulito_id:  \n",
    "        for annotazione in lista_top_annotazioni:\n",
    "            if annotazione in annotazioni_pulito_id[gene] and gene not in nome_lista_geni:\n",
    "                nome_lista_geni.append(gene)\n",
    "    return nome_lista_geni\n",
    "\n",
    "def simboli_con_top_annotazioni (nome_lista_geni, lista_top_annotazioni):\n",
    "    nome_lista_geni= []\n",
    "    for gene in annotazioni_pulito:\n",
    "        for annotazione in lista_top_annotazioni:\n",
    "            if annotazione in annotazioni_pulito[gene] and gene not in nome_lista_geni:\n",
    "                nome_lista_geni.append(gene)\n",
    "    return nome_lista_geni\n",
    "\n",
    "def dizio_vettori_incidenza (dizio_vuoto_incidenza, nome_lista_geni, lista_top_annotazioni):\n",
    "    for gene in nome_lista_geni:\n",
    "        for annot in lista_top_annotazioni:\n",
    "            if gene not in dizio_vuoto_incidenza:\n",
    "                if annot in annotazioni_pulito_id[gene]:\n",
    "                    dizio_vuoto_incidenza[gene]=[1]\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene]= [0]\n",
    "            else:\n",
    "                if annot in annotazioni_pulito_id[gene]:\n",
    "                    dizio_vuoto_incidenza[gene].append(1)\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene].append(0)\n",
    "    matrice_incidenza= pd.DataFrame(dizio_vuoto_incidenza).transpose()\n",
    "    matrice_incidenza.columns= lista_top_annotazioni\n",
    "    return matrice_incidenza\n",
    "\n",
    "def quanti_seeds_in_cluster_g1g2(cluster_map, g1,g2): \n",
    "    conta_seeds_cluster0=0\n",
    "    conta_seeds_cluster1=0\n",
    "    seeds_cluster0= list(cluster_map[cluster_map['cluster']==g1]['idgene'])\n",
    "    seeds_cluster1= list(cluster_map[cluster_map['cluster']==g2]['idgene'])\n",
    "    for gene in seeds_cluster0:\n",
    "        if gene in lista_seeds_unici:\n",
    "            conta_seeds_cluster0 +=1\n",
    "    for gene in seeds_cluster1:\n",
    "        if gene in lista_seeds_unici:\n",
    "            conta_seeds_cluster1 +=1\n",
    "    return print('il numero di seeds nel cluster 0 è: ', conta_seeds_cluster0, 'su', len(seeds_cluster0), 'ovvero il ',  \n",
    "                 conta_seeds_cluster0*100/len(seeds_cluster0),\n",
    "                 '%, \\n mentre nel cluster 1: ', conta_seeds_cluster1, 'su', len(seeds_cluster1),\n",
    "                 'ovvero il ', conta_seeds_cluster1*100/len(seeds_cluster1), '%' )\n",
    "\n",
    "def togli_annot_ridondanti(lista_annotazioni):\n",
    "    copia= []\n",
    "    for an in lista_annotazioni:\n",
    "        if an not in copia:\n",
    "            copia.append(an)\n",
    "        else: \n",
    "            print('annot ridondante: ',an)\n",
    "    lista_annotazioni= copia\n",
    "    return lista_annotazioni\n",
    "\n",
    "def tabella_random_pathways (diz_vuoto, dizio_count, size_campione, numero_campioni, lista_geni_filtrata):\n",
    "    for i in range(1,numero_campioni+1):  \n",
    "        diz_vuoto={}\n",
    "        dizio_count={}\n",
    "        lista_sample_n= np.random.choice(lista_geni_filtrata, size= size_campione, replace=False)  \n",
    "        for simbologene in lista_sample_n:\n",
    "            if simbologene in pathways_pulito_simbolo.keys():\n",
    "                diz_vuoto[simbologene]= pathways_pulito_simbolo[simbologene]\n",
    "        for lista_pathways in diz_vuoto.values():\n",
    "            for pathway in lista_pathways:\n",
    "                if pathway not in dizio_count:\n",
    "                    dizio_count[pathway]=1\n",
    "                else:\n",
    "                    dizio_count[pathway]+=1\n",
    "        series_annotaz_campione= pd.Series(dizio_count)  \n",
    "        valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "        if i==1:\n",
    "            col_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            col_funzioni_random.columns= ['random sample 1']\n",
    "        elif i>1:\n",
    "            tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            tab_funzioni_random.columns= ['random sample ' + str(i)]\n",
    "            col_funzioni_random= col_funzioni_random.join(tab_funzioni_random)\n",
    "    return col_funzioni_random\n",
    "\n",
    "def media_tab_random_pathways (tab_campioni_random, num_campioni_usati):\n",
    "    media_campioni_random= pd.DataFrame(tab_campioni_random).mean(axis=1)\n",
    "    media_campioni_random.sort_values(ascending=False,inplace=True)\n",
    "    media_campioni_random = pd.DataFrame(media_campioni_random)\n",
    "    media_campioni_random.columns= ['media count pathways su ' + str(num_campioni_usati) +' campioni random']\n",
    "    return media_campioni_random\n",
    "\n",
    "def geni_con_top_pathways (nome_lista_geni, lista_top_pathways): \n",
    "    nome_lista_geni= []\n",
    "    for genesimb in pathways_pulito_simbolo:\n",
    "        for pathway in lista_top_pathways:\n",
    "            if pathway in pathways_pulito_simbolo[genesimb] and genesimb not in nome_lista_geni:\n",
    "                nome_lista_geni.append(genesimb)\n",
    "    return nome_lista_geni\n",
    "\n",
    "def dizio_vettori_incidenza_pathways(dizio_vuoto_incidenza, nome_lista_geni, lista_top_pathways):\n",
    "    for gene_s in nome_lista_geni:\n",
    "        for pat in lista_top_pathways:\n",
    "            if gene_s not in dizio_vuoto_incidenza:\n",
    "                if pat in pathways_pulito_simbolo[gene_s]:\n",
    "                    dizio_vuoto_incidenza[gene_s]=[1]\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene_s]= [0]\n",
    "            else:\n",
    "                if pat in pathways_pulito_simbolo[gene_s]:\n",
    "                    dizio_vuoto_incidenza[gene_s].append(1)\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene_s].append(0)\n",
    "    matrice_incidenza= pd.DataFrame(dizio_vuoto_incidenza).transpose()\n",
    "    matrice_incidenza.columns= lista_top_pathways\n",
    "    return matrice_incidenza\n",
    "\n",
    "def quanti_in_cluster_g1g2(cluster_map, g1,g2): \n",
    "    conta_seeds_cluster0=0\n",
    "    conta_seeds_cluster1=0\n",
    "    seeds_cluster0= list(cluster_map[cluster_map['cluster']==g1]['idgene'])\n",
    "    seeds_cluster1= list(cluster_map[cluster_map['cluster']==g2]['idgene'])\n",
    "    for gene in seeds_cluster0:\n",
    "        if gene in lista_seeds_unici:\n",
    "            conta_seeds_cluster0 +=1\n",
    "    for gene in seeds_cluster1:\n",
    "        if gene in lista_seeds_unici:\n",
    "            conta_seeds_cluster1 +=1\n",
    "    return print('il numero di seeds nel cluster 0 è: ', conta_seeds_cluster0, 'su', len(seeds_cluster0), 'ovvero il ',  \n",
    "                 conta_seeds_cluster0*100/len(seeds_cluster0),\n",
    "                 '%, \\n mentre nel cluster 1: ', conta_seeds_cluster1, 'su', len(seeds_cluster1),\n",
    "                 'ovvero il ', conta_seeds_cluster1*100/len(seeds_cluster1), '%' )\n",
    "\n",
    "def quanti_in_cluster_g1g2_pat(cluster_map, g1,g2):\n",
    "    conta_seeds_cluster0=0\n",
    "    conta_seeds_cluster1=0\n",
    "    seeds_cluster0= list(cluster_map[cluster_map['cluster']==g1]['simbolo gene'])\n",
    "    seeds_cluster1= list(cluster_map[cluster_map['cluster']==g2]['simbolo gene'])\n",
    "    for gene in seeds_cluster0:\n",
    "        if gene in simbolo_idgene and simbolo_idgene[gene] in lista_seeds_unici:\n",
    "            conta_seeds_cluster0 +=1\n",
    "    for gene in seeds_cluster1:\n",
    "        if gene in simbolo_idgene and simbolo_idgene[gene] in lista_seeds_unici:\n",
    "            conta_seeds_cluster1 +=1\n",
    "    return print('il numero di seeds nel cluster 0 è: ', conta_seeds_cluster0, 'su', len(seeds_cluster0), 'ovvero il ',  \n",
    "                 conta_seeds_cluster0*100/len(seeds_cluster0),\n",
    "                 '%, \\n mentre nel cluster 1: ', conta_seeds_cluster1, 'su', len(seeds_cluster1),\n",
    "                 'ovvero il ', conta_seeds_cluster1*100/len(seeds_cluster1), '%' )\n",
    "\n",
    "\n",
    "def quanti_in_comune(clustermap1, clustermap2, clust1, clust2):\n",
    "    gruppo1 = clustermap1[clustermap1['cluster']== clust1]\n",
    "    gruppo2 = clustermap2[clustermap2['cluster']== clust2]\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    l=[]\n",
    "    for gene in list(gruppo1['simbolo gene']):\n",
    "        if gene in list(gruppo2['simbolo gene']):\n",
    "            tot_comuni+=1\n",
    "            if gene in simbolo_idgene and simbolo_idgene[gene] in lista_seeds_unici:\n",
    "                seeds_comuni+=1\n",
    "        else:\n",
    "            l.append(gene)\n",
    "    for gene in l:\n",
    "        if gene in simbolo_idgene and simbolo_idgene[gene] in lista_seeds_unici:\n",
    "            seeds_scartati+=1\n",
    "    return 'i geni del gruppo1 sono: ', len(gruppo1), 'quelli del gruppo 2: ', len(gruppo2),'i tot in comune sono', tot_comuni,'in % ', tot_comuni*100/min(len(gruppo1),len(gruppo2)), 'quelli non: ',l, 'quelli seeds in comune', seeds_comuni,'quelli seeds scartati sono: ', seeds_scartati\n",
    "\n",
    "\n",
    "def quanti_in_comune_id(clustermap1, clustermap2, clust1, clust2):\n",
    "    gruppo1 = clustermap1[clustermap1['cluster']== clust1]\n",
    "    gruppo2 = clustermap2[clustermap2['cluster']== clust2]\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    l=[]\n",
    "    for gene in list(gruppo1['idgene']):\n",
    "        if gene in list(gruppo2['idgene']):\n",
    "            tot_comuni+=1\n",
    "            if gene in lista_seeds_unici:\n",
    "                seeds_comuni+=1\n",
    "        else:\n",
    "            l.append(gene)\n",
    "    for gene in l:\n",
    "        if gene in lista_seeds_unici:\n",
    "            seeds_scartati+=1\n",
    "    return 'i geni del gruppo1 sono: ', len(gruppo1), 'quelli del gruppo 2: ', len(gruppo2),'i tot in comune sono', tot_comuni,'in % ', tot_comuni*100/min(len(gruppo1),len(gruppo2)), 'quelli non: ',l, 'quelli seeds in comune', seeds_comuni,'quelli seeds scartati sono: ', seeds_scartati\n",
    "\n",
    "\n",
    "def confronto_annot_patwh(cluster_annot, cluster_path, clust1, clust2):\n",
    "    gruppo1 = cluster_annot[cluster_annot['cluster']== clust1]\n",
    "    gruppo2 = cluster_path[cluster_path['cluster']== clust2]\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    l=[]\n",
    "    for geneid in list(gruppo1['idgene']):\n",
    "        if geneid in idgene_simbolo:   \n",
    "            simb= idgene_simbolo[geneid]  \n",
    "            if simb in list(gruppo2['simbolo gene']):\n",
    "                tot_comuni+=1\n",
    "                if simb in simbolo_idgene and simbolo_idgene[simb] in lista_seeds_unici: \n",
    "                    seeds_comuni+=1     \n",
    "            else:\n",
    "                l.append(geneid)\n",
    "    for geneid in l:\n",
    "        if geneid in lista_seeds_unici:\n",
    "            seeds_scartati+=1\n",
    "    return 'i geni del gruppo1 sono: ', len(gruppo1), 'quelli del gruppo 2: ', len(gruppo2),'i tot in comune sono', tot_comuni,'in % ', tot_comuni*100/min(len(gruppo1),len(gruppo2)), 'quelli non: ', 'quelli seeds in comune', seeds_comuni,'quelli seeds scartati sono: ', seeds_scartati\n",
    "\n",
    "\n",
    "def resoconto_seeds_cluster(cluster_map, n_clusters, soglia):\n",
    "    for i in range(n_clusters):\n",
    "        rossi_malattia=0\n",
    "        rossi_tot=0\n",
    "        cluster_iesimo= cluster_map[cluster_map['cluster']==i]['idgene']\n",
    "        for gene in cluster_iesimo:\n",
    "            if gene in lista_seeds_unici:\n",
    "                rossi_tot+=1\n",
    "                if gene in geni_carboidrati:\n",
    "                    rossi_malattia+=1\n",
    "        if rossi_malattia>= soglia and len(cluster_iesimo)>0:\n",
    "            print(rossi_malattia, 'su', len (cluster_iesimo), 'e i rossi tot ', rossi_tot, ', il', \n",
    "             round(rossi_tot*100/len(cluster_iesimo),1), '%',i)\n",
    "       # else:\n",
    "           # print('cluster nullo')\n",
    "\n",
    "\n",
    "def tabella_random_simboli (diz_vuoto, dizio_count, size_campione, numero_campioni, lista_geni_filtrata):\n",
    "    for i in range(1,numero_campioni+1):  \n",
    "        diz_vuoto={}\n",
    "        dizio_count={}\n",
    "        lista_sample_n= np.random.choice(lista_geni_filtrata, size= size_campione, replace=False)  \n",
    "        for simbologene in lista_sample_n:\n",
    "            if simbologene in annotazioni_pulito.keys():\n",
    "                diz_vuoto[simbologene]= annotazioni_pulito[simbologene]\n",
    "        for lista_funzioni in diz_vuoto.values():\n",
    "            for funzione in lista_funzioni:\n",
    "                if funzione not in dizio_count:\n",
    "                    dizio_count[funzione]=1\n",
    "                else:\n",
    "                    dizio_count[funzione]+=1\n",
    "        series_annotaz_campione= pd.Series(dizio_count)  \n",
    "        valori_descrescenti= series_annotaz_campione.sort_values(ascending=False)\n",
    "        if i==1:\n",
    "            col_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            col_funzioni_random.columns= ['random sample 1']\n",
    "        elif i>1:\n",
    "            tab_funzioni_random= pd.DataFrame(valori_descrescenti)\n",
    "            tab_funzioni_random.columns= ['random sample ' + str(i)]\n",
    "            col_funzioni_random= col_funzioni_random.join(tab_funzioni_random)\n",
    "    return col_funzioni_random\n",
    "\n",
    "def mat_incidenza_annot_simboli (dizio_vuoto_incidenza, lista_simboli_geni, lista_top_annotazioni):\n",
    "    for gene_s in lista_simboli_geni:\n",
    "        for annot in lista_top_annotazioni:\n",
    "            if gene_s not in dizio_vuoto_incidenza:\n",
    "                if annot in annotazioni_pulito[gene_s]:\n",
    "                    dizio_vuoto_incidenza[gene_s]=[1]\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene_s]= [0]\n",
    "            else:\n",
    "                if annot in annotazioni_pulito[gene_s]:\n",
    "                    dizio_vuoto_incidenza[gene_s].append(1)\n",
    "                else:\n",
    "                    dizio_vuoto_incidenza[gene_s].append(0)\n",
    "    matrice_incidenza= pd.DataFrame(dizio_vuoto_incidenza).transpose()\n",
    "    matrice_incidenza.columns= lista_top_annotazioni\n",
    "    return matrice_incidenza\n",
    "\n",
    "def conta_simboli_in_lista(lista_geni_top, lista_seeds_o_malattia):\n",
    "    conta=0\n",
    "    for gene in lista_geni_top:\n",
    "        if ufficiale_simbolo_id[gene] in lista_seeds_o_malattia:\n",
    "            conta+=1\n",
    "    print('ci sono', conta,'seed su',len(lista_geni_top),'cioè il',round(conta*100/len(lista_geni_top),1),'%')\n",
    "\n",
    "def conta_id_in_lista(lista_idgeni_top, lista_seeds_o_malattia):\n",
    "    conta=0\n",
    "    for gene in lista_idgeni_top:\n",
    "        if gene in lista_seeds_o_malattia:\n",
    "            conta+=1\n",
    "    print('ci sono', conta,'seed su',len(lista_idgeni_top),'cioè il',round(conta*100/len(lista_idgeni_top),1),'%')\n",
    "\n",
    "def confronto_annot_path_13458_id(cluster_annot, cluster_path, clust1, clust2, lista_idgeni_malattia, lista_vuota_comuni):\n",
    "    gruppo1 = cluster_annot[cluster_annot['cluster']== clust1]\n",
    "    gruppo2 = cluster_path[cluster_path['cluster']== clust2]\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    geni_malattia=0\n",
    "    for geneid in list(gruppo1['idgene']):\n",
    "        if geneid in idgene_simbolo:   \n",
    "            simb= idgene_simbolo[geneid]   \n",
    "            if simb in list(gruppo2['simbolo gene']):\n",
    "                tot_comuni+=1\n",
    "                lista_vuota_comuni.append(simb)\n",
    "                if simb in simbolo_idgene and simbolo_idgene[simb] in lista_seeds_unici: \n",
    "                    seeds_comuni+=1    \n",
    "                    if simb in simbolo_idgene and simbolo_idgene[simb] in lista_idgeni_malattia:\n",
    "                        geni_malattia+=1\n",
    "    return print('gruppo annot: ', len(gruppo1), ', gruppo path: ', len(gruppo2),\n",
    "                 '\\ntot geni comuni', tot_comuni,',in % ', tot_comuni*100/min(len(gruppo1),len(gruppo2)),\n",
    "                  '\\nquelli seeds in comune', seeds_comuni,'\\nquelli della malattia comuni: ', geni_malattia,\n",
    "                'lunghezza lista comuni è: ',len(lista_vuota_comuni))\n",
    "\n",
    "\n",
    "\n",
    "def confronto_ann_pat_ufficiale_17826(cluster_annot, cluster_path, clust1, clust2, lista_idgeni_malattia,lista_vuota_comuni):\n",
    "    gruppo1 = cluster_annot[cluster_annot['cluster']== clust1]\n",
    "    gruppo2 = cluster_path[cluster_path['cluster']== clust2]\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    geni_malattia=0\n",
    "    for gene_simb in list(gruppo1['simbolo gene']):\n",
    "        if gene_simb in annotazioni_pulito:   \n",
    "            if gene_simb in list(gruppo2['simbolo gene']):\n",
    "                tot_comuni+=1\n",
    "                lista_vuota_comuni.append(gene_simb)\n",
    "                if gene_simb in simbolo_idgene and simbolo_idgene[gene_simb] in lista_seeds_unici: #tanto tutti i seeds hanno l'ID, stanno\n",
    "                    seeds_comuni+=1     # quindi in simbolo_idgene\n",
    "                    if gene_simb in simbolo_idgene and simbolo_idgene[gene_simb] in lista_idgeni_malattia:\n",
    "                        geni_malattia+=1\n",
    "    return print('gruppo annot: ', len(gruppo1), ', gruppo path: ', len(gruppo2),\n",
    "                 '\\ntot geni comuni', tot_comuni,',in % ', tot_comuni*100/min(len(gruppo1),len(gruppo2)),\n",
    "                  '\\nquelli seeds in comune', seeds_comuni,'\\nquelli della malattia comuni: ', geni_malattia,\n",
    "                  '\\nlista geni comuni è:',len(lista_vuota_comuni))\n",
    "\n",
    "def campione_simboli_random (diz_vuoto, n_campione, filtro_simboli):\n",
    "    lista_sample_n= np.random.choice(filtro_simboli, size=n_campione, replace=False)\n",
    "    for simb in lista_sample_n:\n",
    "        if simb in annotazioni_pulito:\n",
    "            diz_vuoto[simb]= annotazioni_pulito[simb]\n",
    "    return diz_vuoto\n",
    "\n",
    "def togli_nan_da_lista_simboli (lista_simboli):\n",
    "    for elemento in lista_simboli:\n",
    "        if type(elemento)== float and math.isnan(elemento)==True:\n",
    "            lista_simboli.remove(elemento)\n",
    "    return lista_simboli\n",
    "\n",
    "def separa_seed_tra_comuni (non_malattia_comuni, simboli_carbo_comuni, idgene_carbo_comuni, tot_simb_comuni):\n",
    "    for simbolo in tot_simb_comuni:\n",
    "        if simbolo in simboli_geni_carboidrati:\n",
    "            simboli_carbo_comuni.append(simbolo)\n",
    "            idgene_carbo_comuni.append(simbolo_idgene[simbolo])\n",
    "        else:\n",
    "            non_malattia_comuni.append(simbolo)\n",
    "    return non_malattia_comuni, simboli_carbo_comuni, idgene_carbo_comuni\n",
    "\n",
    "def geni_adiacenti (lista_vuota_adiacenti, id_carbo_comuni):\n",
    "    for nodo in id_carbo_comuni:\n",
    "        for adiacente in interattoma_completo[nodo]:\n",
    "            if adiacente not in lista_vuota_adiacenti:\n",
    "                lista_vuota_adiacenti.append(adiacente)\n",
    "    return((lista_vuota_adiacenti))\n",
    "\n",
    "\n",
    "def resoconto_seeds_cluster(cluster_map, n_clusters, soglia, geni_malattia):\n",
    "    for i in range(n_clusters):\n",
    "        rossi_malattia=0\n",
    "        rossi_tot=0\n",
    "        cluster_iesimo= cluster_map[cluster_map['cluster']==i]['idgene']\n",
    "        for gene in cluster_iesimo:\n",
    "            if gene in lista_seeds_unici:\n",
    "                rossi_tot+=1\n",
    "                if gene in geni_malattia:\n",
    "                    rossi_malattia+=1\n",
    "        if rossi_malattia>= soglia and len(cluster_iesimo)>0:\n",
    "            print(rossi_malattia, 'su', len (cluster_iesimo), 'e i rossi tot ', rossi_tot, ', il', \n",
    "             round(rossi_tot*100/len(cluster_iesimo),1), '%',i)\n",
    "       # else:\n",
    "           # print('cluster nullo')\n",
    "        \n",
    "def resoconto_seeds_cluster_pat(cluster_map, n_clusters, soglia, geni_malattia):\n",
    "    for i in range(n_clusters):\n",
    "        rossi_malattia=0\n",
    "        rossi_tot=0\n",
    "        cluster_iesimo= cluster_map[cluster_map['cluster']==i]['simbolo gene']\n",
    "        for gene in cluster_iesimo:\n",
    "            if ufficiale_simbolo_id[gene] in lista_seeds_unici:\n",
    "                rossi_tot+=1\n",
    "                if ufficiale_simbolo_id[gene] in geni_malattia:\n",
    "                    rossi_malattia+=1\n",
    "        if rossi_malattia>= soglia and len(cluster_iesimo)>0:\n",
    "            print(rossi_malattia, 'su', len (cluster_iesimo), 'e i rossi tot ', rossi_tot, ', il', \n",
    "             round(rossi_tot*100/len(cluster_iesimo),1), '%',i)\n",
    "        #else:\n",
    "            #print('cluster nullo')\n",
    "            \n",
    "            \n",
    "            \n",
    "def campione_simboli_random (diz_vuoto, n_campione, filtro_simboli):\n",
    "    lista_sample_n= np.random.choice(filtro_simboli, size=n_campione, replace=False)\n",
    "    for simb in lista_sample_n:\n",
    "        if simb in annotazioni_pulito:\n",
    "            diz_vuoto[simb]= annotazioni_pulito[simb]\n",
    "    return diz_vuoto\n",
    "\n",
    "def togli_nan_da_lista_simboli (lista_simboli):\n",
    "    for elemento in lista_simboli:\n",
    "        if type(elemento)== float and math.isnan(elemento)==True:\n",
    "            lista_simboli.remove(elemento)\n",
    "    return lista_simboli\n",
    "\n",
    "def separa_seed_tra_comuni (non_malattia_comuni, simboli_carbo_comuni, idgene_carbo_comuni, tot_simb_comuni):\n",
    "    for simbolo in tot_simb_comuni:\n",
    "        if simbolo in simboli_geni_carboidrati:\n",
    "            simboli_carbo_comuni.append(simbolo)\n",
    "            idgene_carbo_comuni.append(simbolo_idgene[simbolo])\n",
    "        else:\n",
    "            non_malattia_comuni.append(simbolo)\n",
    "    return non_malattia_comuni, simboli_carbo_comuni, idgene_carbo_comuni\n",
    "\n",
    "def geni_adiacenti (lista_vuota_adiacenti, id_carbo_comuni):\n",
    "    for nodo in id_carbo_comuni:\n",
    "        for adiacente in interattoma_completo[nodo]:\n",
    "            if adiacente not in lista_vuota_adiacenti:\n",
    "                lista_vuota_adiacenti.append(adiacente)\n",
    "    return((lista_vuota_adiacenti))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hide_input": true
   },
   "outputs": [],
   "source": [
    "#### Functions for comparison after the clustering\n",
    "\n",
    "def confronto_13_bests(best_group_ann, best_group_pat, lista_idgeni_malattia, lista_vuota_comuni):\n",
    "    tot_comuni=0\n",
    "    seeds_comuni=0\n",
    "    seeds_scartati=0\n",
    "    geni_malattia=0\n",
    "    for geneid in best_group_ann:\n",
    "        if geneid in idgene_simbolo:   \n",
    "            if simb in best_group_pat:\n",
    "                tot_comuni+=1\n",
    "                lista_vuota_comuni.append(simb)\n",
    "                if simb in simbolo_idgene and simbolo_idgene[simb] in lista_seeds_unici: \n",
    "                    if simb in simbolo_idgene and simbolo_idgene[simb] in lista_idgeni_malattia:\n",
    "                        geni_malattia+=1\n",
    "    return print('gruppo annot: ', len(best_group_ann), ', gruppo path: ', len(best_group_pat),\n",
    "                 '\\ntot geni comuni', tot_comuni,',in % ', tot_comuni*100/min(len(best_group_ann),len(best_group_pat)),\n",
    "                  '\\nquelli seeds in comune', seeds_comuni,'\\nquelli della malattia comuni: ', geni_malattia,\n",
    "                'lunghezza lista comuni è: ',len(lista_vuota_comuni))\n",
    "\n",
    "def rapporto_idseeds_gruppo(lista_idgeni_top, lista_seeds_o_malattia):\n",
    "    conta=0\n",
    "    for gene in lista_idgeni_top:\n",
    "        if gene in lista_seeds_o_malattia:\n",
    "            conta+=1\n",
    "    return round(conta*100/len(lista_idgeni_top),1)\n",
    "\n",
    "def rapporto_simbseeds_gruppo(lista_simb_top, lista_seeds_o_malattia):\n",
    "    conta=0\n",
    "    for gene in lista_simb_top:\n",
    "        if ufficiale_simbolo_id[gene] in lista_seeds_o_malattia:\n",
    "            conta+=1\n",
    "    return round(conta*100/len(lista_simb_top),1)\n",
    "\n",
    "def quanti_seed_totali(lista_idgeni_top, lista_seeds):\n",
    "    conta=0\n",
    "    for gene in lista_idgeni_top:\n",
    "        if gene in lista_seeds:\n",
    "            conta+=1\n",
    "    return conta\n",
    "\n",
    "def quanti_seed_simb_tot(lista_simb_top, lista_seeds):\n",
    "    conta=0\n",
    "    try:\n",
    "        for gene in lista_simb_top:\n",
    "            if ufficiale_simbolo_id[gene] in lista_seeds:\n",
    "                conta+=1\n",
    "    except KeyError:\n",
    "        pass\n",
    "    return conta\n",
    "\n",
    "def uff_trova_chiave_da(dizionario,valore): \n",
    "    for key in dizionario:\n",
    "        if dizionario[key]== valore:\n",
    "            return key\n",
    "\n",
    "def uff_resoconto_seeds_cluster(cluster_map, n_clusters, soglia, geni_malattia):\n",
    "    for i in range(n_clusters):\n",
    "        rossi_malattia=0\n",
    "        rossi_tot=0\n",
    "        cluster_iesimo= cluster_map[cluster_map['cluster']==i]['idgene']\n",
    "        for gene in cluster_iesimo:\n",
    "            if gene in lista_seeds_unici:\n",
    "                rossi_tot+=1\n",
    "                if gene in geni_malattia:\n",
    "                    rossi_malattia+=1\n",
    "        if rossi_malattia>= soglia and len(cluster_iesimo)>0:\n",
    "            print(rossi_malattia, 'su', len (cluster_iesimo), 'e i rossi tot ', rossi_tot, ', il', \n",
    "             round(rossi_tot*100/len(cluster_iesimo),1), '%',i)\n",
    "       # else:\n",
    "           # print('cluster nullo')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Cleaning and Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Interactom dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('C:\\\\Users\\\\feder\\\\jupyter\\\\tesi network medicine')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ID_1</th>\n",
       "      <th>gene_ID_2</th>\n",
       "      <th>gene_symbol_1</th>\n",
       "      <th>gene_symbol_2</th>\n",
       "      <th>sources</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>310</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ANXA7</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1026</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_ID_1      gene_ID_2   gene_symbol_1   gene_symbol_2    sources\n",
       "0              1         310            A1BG          ANXA7  signaling\n",
       "1              1        1026            A1BG         CDKN1A  signaling"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#os.getcwd()\n",
    "interattoma= pd.read_table('Interactome.TSV')\n",
    "interattoma.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID gene uscita</th>\n",
       "      <th>ID gene entrata</th>\n",
       "      <th>simbolo_uscita</th>\n",
       "      <th>simbolo_entrata</th>\n",
       "      <th>sources</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>310</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ANXA7</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1026</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>2886</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GRB7</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>6606</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>6622</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>SNCA</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>7083</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>TK1</td>\n",
       "      <td>signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>10321</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>CRISP3</td>\n",
       "      <td>literature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>259</td>\n",
       "      <td>A2M</td>\n",
       "      <td>AMBP</td>\n",
       "      <td>literature,signaling</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID gene uscita  ID gene entrata simbolo_uscita simbolo_entrata  \\\n",
       "0               1              310           A1BG           ANXA7   \n",
       "1               1             1026           A1BG          CDKN1A   \n",
       "2               1             2886           A1BG            GRB7   \n",
       "3               1             6606           A1BG            SMN1   \n",
       "4               1             6622           A1BG            SNCA   \n",
       "5               1             7083           A1BG             TK1   \n",
       "6               1            10321           A1BG          CRISP3   \n",
       "7               2              259            A2M            AMBP   \n",
       "\n",
       "                sources  \n",
       "0             signaling  \n",
       "1             signaling  \n",
       "2             signaling  \n",
       "3             signaling  \n",
       "4             signaling  \n",
       "5             signaling  \n",
       "6            literature  \n",
       "7  literature,signaling  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interattoma.columns=['ID gene uscita', 'ID gene entrata', 'simbolo_uscita', 'simbolo_entrata', 'sources']\n",
    "interattoma.head(8)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create dictionary which links each gene (ID) all the genes it is linked to"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz={}  \n",
    "for i in range (0, 141272):\n",
    "    gene_i= interattoma.loc[i,'ID gene uscita']\n",
    "    valore= interattoma.loc[i,'ID gene entrata']\n",
    "    l=[]\n",
    "    if gene_i not in diz:\n",
    "        l.append(valore)\n",
    "        diz[gene_i]= l\n",
    "    else:\n",
    "        diz[gene_i].append(valore)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotations dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene symbol</th>\n",
       "      <th>Annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNAJC25-GNG10</td>\n",
       "      <td>7186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IGKV3-7</td>\n",
       "      <td>2377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IGKV3-7</td>\n",
       "      <td>6955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IGKV1D-42</td>\n",
       "      <td>2377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IGKV1D-42</td>\n",
       "      <td>6955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>IGLV4-69</td>\n",
       "      <td>2250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene symbol  Annotation\n",
       "0  DNAJC25-GNG10        7186\n",
       "1        IGKV3-7        2377\n",
       "2        IGKV3-7        6955\n",
       "3      IGKV1D-42        2377\n",
       "4      IGKV1D-42        6955\n",
       "5       IGLV4-69        2250"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotazioni= pd.read_excel('GO_ann.xlsx')\n",
    "annotazioni.head(6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create dictionary which gives the list of annotations for each gene ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz_annotazioni={}\n",
    "for i in range (0, 165774):\n",
    "    simbolo_gene = annotazioni.iloc[i,0]\n",
    "    funzione= annotazioni.iloc[i,1]\n",
    "    lis= []\n",
    "    if simbolo_gene not in diz_annotazioni:\n",
    "        lis.append(funzione)\n",
    "        diz_annotazioni[simbolo_gene]= lis\n",
    "    else:\n",
    "        diz_annotazioni[simbolo_gene].append(funzione)\n",
    "diz_annotazioni"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotazioni_pulito={}  #cleaned dictionary, without any redoundancy\n",
    "for i in range (0, 165774):\n",
    "    simbolo_gene = annotazioni.iloc[i,0]\n",
    "    funzione= annotazioni.iloc[i,1]\n",
    "    lis= []\n",
    "    if simbolo_gene not in annotazioni_pulito:\n",
    "        lis.append(funzione)\n",
    "        annotazioni_pulito[simbolo_gene]= lis\n",
    "    elif simbolo_gene in annotazioni_pulito and funzione not in annotazioni_pulito[simbolo_gene] :\n",
    "        annotazioni_pulito[simbolo_gene].append(funzione)\n",
    "annotazioni_pulito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Diseases</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 590</th>\n",
       "      <th>Unnamed: 591</th>\n",
       "      <th>Unnamed: 592</th>\n",
       "      <th>Unnamed: 593</th>\n",
       "      <th>Unnamed: 594</th>\n",
       "      <th>Unnamed: 595</th>\n",
       "      <th>Unnamed: 596</th>\n",
       "      <th>Unnamed: 597</th>\n",
       "      <th>Unnamed: 598</th>\n",
       "      <th>Unnamed: 599</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>adrenal gland diseases</td>\n",
       "      <td>3758</td>\n",
       "      <td>215</td>\n",
       "      <td>3762</td>\n",
       "      <td>1589</td>\n",
       "      <td>1585</td>\n",
       "      <td>6770</td>\n",
       "      <td>2516</td>\n",
       "      <td>6557</td>\n",
       "      <td>5573</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>alzheimer disease</td>\n",
       "      <td>5663</td>\n",
       "      <td>23036</td>\n",
       "      <td>348</td>\n",
       "      <td>5664</td>\n",
       "      <td>55103</td>\n",
       "      <td>10452</td>\n",
       "      <td>1191</td>\n",
       "      <td>2629</td>\n",
       "      <td>2041</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>amino acid metabolism inborn errors</td>\n",
       "      <td>445</td>\n",
       "      <td>383</td>\n",
       "      <td>2109</td>\n",
       "      <td>3815</td>\n",
       "      <td>11234</td>\n",
       "      <td>388552</td>\n",
       "      <td>4286</td>\n",
       "      <td>275</td>\n",
       "      <td>2110</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>amyotrophic lateral sclerosis</td>\n",
       "      <td>6647</td>\n",
       "      <td>1639</td>\n",
       "      <td>6311</td>\n",
       "      <td>57679</td>\n",
       "      <td>10133</td>\n",
       "      <td>2521</td>\n",
       "      <td>998</td>\n",
       "      <td>22920</td>\n",
       "      <td>203228</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>anemia aplastic</td>\n",
       "      <td>3458</td>\n",
       "      <td>6125</td>\n",
       "      <td>6135</td>\n",
       "      <td>7015</td>\n",
       "      <td>2187</td>\n",
       "      <td>6229</td>\n",
       "      <td>2177</td>\n",
       "      <td>2175</td>\n",
       "      <td>2176</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Diseases  Genes  Unnamed: 2  Unnamed: 3  \\\n",
       "0               adrenal gland diseases   3758         215        3762   \n",
       "1                    alzheimer disease   5663       23036         348   \n",
       "2  amino acid metabolism inborn errors    445         383        2109   \n",
       "3        amyotrophic lateral sclerosis   6647        1639        6311   \n",
       "4                      anemia aplastic   3458        6125        6135   \n",
       "\n",
       "   Unnamed: 4  Unnamed: 5  Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  \\\n",
       "0        1589        1585        6770        2516        6557        5573   \n",
       "1        5664       55103       10452        1191        2629        2041   \n",
       "2        3815       11234      388552        4286         275        2110   \n",
       "3       57679       10133        2521         998       22920      203228   \n",
       "4        7015        2187        6229        2177        2175        2176   \n",
       "\n",
       "       ...       Unnamed: 590  Unnamed: 591  Unnamed: 592  Unnamed: 593  \\\n",
       "0      ...                NaN           NaN           NaN           NaN   \n",
       "1      ...                NaN           NaN           NaN           NaN   \n",
       "2      ...                NaN           NaN           NaN           NaN   \n",
       "3      ...                NaN           NaN           NaN           NaN   \n",
       "4      ...                NaN           NaN           NaN           NaN   \n",
       "\n",
       "   Unnamed: 594  Unnamed: 595  Unnamed: 596  Unnamed: 597  Unnamed: 598  \\\n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "1           NaN           NaN           NaN           NaN           NaN   \n",
       "2           NaN           NaN           NaN           NaN           NaN   \n",
       "3           NaN           NaN           NaN           NaN           NaN   \n",
       "4           NaN           NaN           NaN           NaN           NaN   \n",
       "\n",
       "   Unnamed: 599  \n",
       "0           NaN  \n",
       "1           NaN  \n",
       "2           NaN  \n",
       "3           NaN  \n",
       "4           NaN  \n",
       "\n",
       "[5 rows x 600 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seeds= pd.read_excel('seeds.xlsx')\n",
    "seeds.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Associate gene's symbol to each gene ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "idgene_simbolo={}  #ho dizionario in cui a ogni ID di un gene ho associato il suo simbolo\n",
    "simbolo_idgene={}\n",
    "for indice in range (0,141272):\n",
    "    riga = interattoma.iloc[indice,:].values #trasforma ogni riga in lista\n",
    "    id_uscita = riga[0]  #prende la colonna in posizione 0, la prima\n",
    "    id_entrata= riga[1]\n",
    "    simbolo_uscita= riga[2] #prende collona in posiz 2\n",
    "    simbolo_entrata= riga[3]\n",
    "    if id_uscita not in idgene_simbolo:\n",
    "        idgene_simbolo[id_uscita]= simbolo_uscita\n",
    "        simbolo_idgene[simbolo_uscita]= id_uscita\n",
    "    if id_entrata not in idgene_simbolo:\n",
    "        idgene_simbolo[id_entrata]= simbolo_entrata\n",
    "        simbolo_idgene[simbolo_entrata]= id_entrata\n",
    "idgene_simbolo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "for simbolo in annotazioni_pulito:\n",
    "    if simbolo not in simbolo_idgene:\n",
    "        simbolo_idgene[simbolo]= ''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create annotazioni_pulito_id, which has the list of annotations for each gene ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotazioni_pulito  # es: 'A1BG': [24, 543, 221]\n",
    "idgene_simbolo   # es: 1: 'A1BG'\n",
    "annotazioni_pulito_id={} "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for gene_simbolo in idgene_simbolo.values():  \n",
    "    for symbol in annotazioni_pulito.keys():  \n",
    "        if gene_simbolo== symbol:  \n",
    "            chiave_id= trova_chiave_da(idgene_simbolo, gene_simbolo)  \n",
    "            annotazioni_pulito_id[chiave_id]= annotazioni_pulito[symbol] \n",
    "annotazioni_pulito_id     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_funzioni_random= count_campione_geni(random,count_funzioni_random)\n",
    "count_funzioni_random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Automating the whole process of creating a column based on a new random sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "random_sample={}\n",
    "count_funzioni_random={}\n",
    "annotaz_malattia_carboidrati={}\n",
    "dizio_count={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 3758.,   215.,  3762., ...,    nan,    nan,    nan],\n",
       "       [ 5663., 23036.,   348., ...,    nan,    nan,    nan],\n",
       "       [  445.,   383.,  2109., ...,    nan,    nan,    nan],\n",
       "       ...,\n",
       "       [ 4340.,  1121.,  4276., ...,    nan,    nan,    nan],\n",
       "       [51438.,  7247.,   652., ...,    nan,    nan,    nan],\n",
       "       [ 4340.,  4276., 80271., ...,    nan,    nan,    nan]])"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "array_seeds= seeds.iloc[:,1:].values\n",
    "array_seeds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2843"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lista_geni_seeds=[]\n",
    "for lista in array_seeds:\n",
    "    for gene_seed in lista:\n",
    "        if math.isnan(gene_seed)== False:\n",
    "            lista_geni_seeds.append(int(gene_seed))\n",
    "len(lista_geni_seeds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "series_geni_seeds= pd.Series(lista_geni_seeds)\n",
    "lista_seeds_unici= series_geni_seeds.unique()   #1536 seeds unici su 2843 in tutte le malattie (un gene puo comparire in piu)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3118    15\n",
       "3119    15\n",
       "3117    10\n",
       "4276    10\n",
       "4855     8\n",
       "dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series_geni_seeds.value_counts().head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### number of annotations for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "dizio_annotazioni_seeds={}\n",
    "count_annotazioni_seeds={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "unici= list(lista_seeds_unici)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotazioni_geni_unici=[]\n",
    "for gene in unici:\n",
    "    if gene in annotazioni_pulito_id:\n",
    "        for annotaz in annotazioni_pulito_id[gene]:\n",
    "            annotazioni_geni_unici.append(annotaz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_annotaz_unici={}\n",
    "for gene in annotazioni_geni_unici:\n",
    "    if gene not in count_annotaz_unici:\n",
    "        count_annotaz_unici[gene]=1      #'AGPS'\n",
    "    else:\n",
    "        count_annotaz_unici[gene]+=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22976"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(annotazioni_geni_unici)  #22976"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_annotaz_unici= pd.Series(count_annotaz_unici).sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count annotazione in geni seeds unici</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45944</th>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       count annotazione in geni seeds unici\n",
       "45944                                    188\n",
       "122                                      134"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tabella_annotazioni_unici= pd.DataFrame(count_annotaz_unici)\n",
    "tabella_annotazioni_unici.columns= ['count annotazione in geni seeds unici']\n",
    "tabella_annotazioni_unici.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz_count_geni_seeds={}\n",
    "for gene in lista_geni_seeds:\n",
    "    if gene not in diz_count_geni_seeds:\n",
    "        diz_count_geni_seeds[gene]=1\n",
    "    else:\n",
    "        diz_count_geni_seeds[gene]+=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>conta di ciascun gene seed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3118</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3119</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      conta di ciascun gene seed\n",
       "3118                          15\n",
       "3119                          15"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diz_count_geni_seeds= pd.Series(diz_count_geni_seeds).sort_values(ascending=False)\n",
    "diz_count_geni_seeds= pd.DataFrame(diz_count_geni_seeds)\n",
    "diz_count_geni_seeds.columns= ['conta di ciascun gene seed']\n",
    "diz_count_geni_seeds.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "lista_geni_senza_simbolo=[]\n",
    "for simbolo in annotazioni_pulito:\n",
    "    if simbolo not in idgene_simbolo.values():\n",
    "        lista_geni_senza_simbolo.append(simbolo)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### List of genes without known annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "simboli_senza_annotazioni=[]\n",
    "geni_non_interagenti=[]\n",
    "for simb in simbolo_idgene:\n",
    "    if simb not in annotazioni_pulito:\n",
    "        simboli_senza_annotazioni.append(simb)\n",
    "    if simbolo_idgene[simb] == '':\n",
    "        geni_non_interagenti.append(simb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Automating creation of total table where each column is the result of a random sample. Creating incidence table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "random_sample={}\n",
    "count_funzioni_random={}\n",
    "annotaz_malattia_carboidrati={}\n",
    "dizio_count={}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Working on 2nd set of annotations (Pathways)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathways= pd.read_excel('Pathways_KEGG.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "186\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pathway</th>\n",
       "      <th>gene 1</th>\n",
       "      <th>gene 2</th>\n",
       "      <th>gene 3</th>\n",
       "      <th>gene 4</th>\n",
       "      <th>gene 5</th>\n",
       "      <th>gene 6</th>\n",
       "      <th>gene 7</th>\n",
       "      <th>gene 8</th>\n",
       "      <th>gene 9</th>\n",
       "      <th>...</th>\n",
       "      <th>gene 380</th>\n",
       "      <th>gene 381</th>\n",
       "      <th>gene 382</th>\n",
       "      <th>gene 383</th>\n",
       "      <th>gene 384</th>\n",
       "      <th>gene 385</th>\n",
       "      <th>gene 386</th>\n",
       "      <th>gene 387</th>\n",
       "      <th>gene 388</th>\n",
       "      <th>gene 389</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GLYCOLYSIS_GLUCONEOGENESIS</td>\n",
       "      <td>ACSS2</td>\n",
       "      <td>GCK</td>\n",
       "      <td>PGK2</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>PDHB</td>\n",
       "      <td>PDHA1</td>\n",
       "      <td>PDHA2</td>\n",
       "      <td>PGM2</td>\n",
       "      <td>TPI1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CITRATE_CYCLE_TCA_CYCLE</td>\n",
       "      <td>IDH3B</td>\n",
       "      <td>DLST</td>\n",
       "      <td>PCK2</td>\n",
       "      <td>CS</td>\n",
       "      <td>PDHB</td>\n",
       "      <td>PCK1</td>\n",
       "      <td>PDHA1</td>\n",
       "      <td>LOC642502</td>\n",
       "      <td>PDHA2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PENTOSE_PHOSPHATE_PATHWAY</td>\n",
       "      <td>RPE</td>\n",
       "      <td>RPIA</td>\n",
       "      <td>PGM2</td>\n",
       "      <td>PGLS</td>\n",
       "      <td>PRPS2</td>\n",
       "      <td>FBP2</td>\n",
       "      <td>PFKM</td>\n",
       "      <td>PFKL</td>\n",
       "      <td>TALDO1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS</td>\n",
       "      <td>UGT1A10</td>\n",
       "      <td>UGT1A8</td>\n",
       "      <td>RPE</td>\n",
       "      <td>UGT1A7</td>\n",
       "      <td>UGT1A6</td>\n",
       "      <td>UGT2B28</td>\n",
       "      <td>UGT1A5</td>\n",
       "      <td>CRYL1</td>\n",
       "      <td>UGDH</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FRUCTOSE_AND_MANNOSE_METABOLISM</td>\n",
       "      <td>MPI</td>\n",
       "      <td>PMM2</td>\n",
       "      <td>PMM1</td>\n",
       "      <td>FBP2</td>\n",
       "      <td>PFKM</td>\n",
       "      <td>GMDS</td>\n",
       "      <td>PFKFB4</td>\n",
       "      <td>PFKL</td>\n",
       "      <td>MTMR6</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 390 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    pathway   gene 1  gene 2 gene 3  gene 4  \\\n",
       "0                GLYCOLYSIS_GLUCONEOGENESIS    ACSS2     GCK   PGK2    PGK1   \n",
       "1                   CITRATE_CYCLE_TCA_CYCLE    IDH3B    DLST   PCK2      CS   \n",
       "2                 PENTOSE_PHOSPHATE_PATHWAY      RPE    RPIA   PGM2    PGLS   \n",
       "3  PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS  UGT1A10  UGT1A8    RPE  UGT1A7   \n",
       "4           FRUCTOSE_AND_MANNOSE_METABOLISM      MPI    PMM2   PMM1    FBP2   \n",
       "\n",
       "   gene 5   gene 6  gene 7     gene 8  gene 9   ...    gene 380 gene 381  \\\n",
       "0    PDHB    PDHA1   PDHA2       PGM2    TPI1   ...         NaN      NaN   \n",
       "1    PDHB     PCK1   PDHA1  LOC642502   PDHA2   ...         NaN      NaN   \n",
       "2   PRPS2     FBP2    PFKM       PFKL  TALDO1   ...         NaN      NaN   \n",
       "3  UGT1A6  UGT2B28  UGT1A5      CRYL1    UGDH   ...         NaN      NaN   \n",
       "4    PFKM     GMDS  PFKFB4       PFKL   MTMR6   ...         NaN      NaN   \n",
       "\n",
       "  gene 382 gene 383 gene 384 gene 385 gene 386 gene 387 gene 388 gene 389  \n",
       "0      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "1      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "2      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "3      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "4      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "\n",
       "[5 rows x 390 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(pathways))\n",
    "pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathways_array= pathways.iloc[:,:].values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Costruisco pathways_pulito_simbolo, che associa a ogni simbolo lista dei pathways che ha"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b> Creating incidence matrix for pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACSS2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'GCK': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS',\n",
       "  'MATURITY_ONSET_DIABETES_OF_THE_YOUNG'],\n",
       " 'PGK2': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'PGK1': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'PDHB': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'PDHA1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'PDHA2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'PGM2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'TPI1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'INOSITOL_PHOSPHATE_METABOLISM'],\n",
       " 'ACSS1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'FBP1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'ADH1B': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'HK2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS'],\n",
       " 'ADH1C': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'HK1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS'],\n",
       " 'HK3': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS'],\n",
       " 'ADH4': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'PGAM2': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'ADH5': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'PGAM1': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'ADH1A': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ALDOC': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'ALDH7A1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'LDHAL6B': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'PKLR': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PURINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS',\n",
       "  'MATURITY_ONSET_DIABETES_OF_THE_YOUNG'],\n",
       " 'LDHAL6A': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'ENO1': ['GLYCOLYSIS_GLUCONEOGENESIS', 'RNA_DEGRADATION'],\n",
       " 'PKM2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PURINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'TYPE_II_DIABETES_MELLITUS'],\n",
       " 'PFKP': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM'],\n",
       " 'BPGM': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'PCK2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'PCK1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'ALDH1B1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'ALDH2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'ALDH3A1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'AKR1A1': ['GLYCOLYSIS_GLUCONEOGENESIS', 'GLYCEROLIPID_METABOLISM'],\n",
       " 'FBP2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'PFKM': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM'],\n",
       " 'PFKL': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM'],\n",
       " 'LDHC': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'GAPDH': ['GLYCOLYSIS_GLUCONEOGENESIS', 'ALZHEIMERS_DISEASE'],\n",
       " 'ENO3': ['GLYCOLYSIS_GLUCONEOGENESIS', 'RNA_DEGRADATION'],\n",
       " 'ENO2': ['GLYCOLYSIS_GLUCONEOGENESIS', 'RNA_DEGRADATION'],\n",
       " 'PGAM4': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'ADH7': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ADH6': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'LDHB': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'ALDH1A3': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ALDH3B1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ALDH3B2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ALDH9A1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'ALDH3A2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'GALM': ['GLYCOLYSIS_GLUCONEOGENESIS'],\n",
       " 'ALDOA': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'DLD': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'PYRUVATE_METABOLISM'],\n",
       " 'DLAT': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PYRUVATE_METABOLISM'],\n",
       " 'ALDOB': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'G6PC2': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'LDHA': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'G6PC': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'PGM1': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GPI': ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " nan: ['GLYCOLYSIS_GLUCONEOGENESIS',\n",
       "  'CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'FATTY_ACID_METABOLISM',\n",
       "  'STEROID_BIOSYNTHESIS',\n",
       "  'PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'N_GLYCAN_BIOSYNTHESIS',\n",
       "  'OTHER_GLYCAN_DEGRADATION',\n",
       "  'O_GLYCAN_BIOSYNTHESIS',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'INOSITOL_PHOSPHATE_METABOLISM',\n",
       "  'GLYCOSYLPHOSPHATIDYLINOSITOL_GPI_ANCHOR_BIOSYNTHESIS',\n",
       "  'GLYCEROPHOSPHOLIPID_METABOLISM',\n",
       "  'ETHER_LIPID_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'ALPHA_LINOLENIC_ACID_METABOLISM',\n",
       "  'SPHINGOLIPID_METABOLISM',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'ONE_CARBON_POOL_BY_FOLATE',\n",
       "  'RIBOFLAVIN_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
       "  'FOLATE_BIOSYNTHESIS',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'TERPENOID_BACKBONE_BIOSYNTHESIS',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION',\n",
       "  'NITROGEN_METABOLISM',\n",
       "  'SULFUR_METABOLISM',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
       "  'ABC_TRANSPORTERS',\n",
       "  'RIBOSOME',\n",
       "  'RNA_DEGRADATION',\n",
       "  'RNA_POLYMERASE',\n",
       "  'BASAL_TRANSCRIPTION_FACTORS',\n",
       "  'DNA_REPLICATION',\n",
       "  'SPLICEOSOME',\n",
       "  'PROTEASOME',\n",
       "  'PROTEIN_EXPORT',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR',\n",
       "  'MISMATCH_REPAIR',\n",
       "  'HOMOLOGOUS_RECOMBINATION',\n",
       "  'NON_HOMOLOGOUS_END_JOINING',\n",
       "  'MAPK_SIGNALING_PATHWAY',\n",
       "  'ERBB_SIGNALING_PATHWAY',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'PHOSPHATIDYLINOSITOL_SIGNALING_SYSTEM',\n",
       "  'NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION',\n",
       "  'CELL_CYCLE',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'P53_SIGNALING_PATHWAY',\n",
       "  'UBIQUITIN_MEDIATED_PROTEOLYSIS',\n",
       "  'SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT',\n",
       "  'REGULATION_OF_AUTOPHAGY',\n",
       "  'LYSOSOME',\n",
       "  'ENDOCYTOSIS',\n",
       "  'PEROXISOME',\n",
       "  'MTOR_SIGNALING_PATHWAY',\n",
       "  'APOPTOSIS',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'WNT_SIGNALING_PATHWAY',\n",
       "  'DORSO_VENTRAL_AXIS_FORMATION',\n",
       "  'NOTCH_SIGNALING_PATHWAY',\n",
       "  'HEDGEHOG_SIGNALING_PATHWAY',\n",
       "  'TGF_BETA_SIGNALING_PATHWAY',\n",
       "  'AXON_GUIDANCE',\n",
       "  'VEGF_SIGNALING_PATHWAY',\n",
       "  'FOCAL_ADHESION',\n",
       "  'ECM_RECEPTOR_INTERACTION',\n",
       "  'CELL_ADHESION_MOLECULES_CAMS',\n",
       "  'ADHERENS_JUNCTION',\n",
       "  'TIGHT_JUNCTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'COMPLEMENT_AND_COAGULATION_CASCADES',\n",
       "  'ANTIGEN_PROCESSING_AND_PRESENTATION',\n",
       "  'RENIN_ANGIOTENSIN_SYSTEM',\n",
       "  'TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY',\n",
       "  'NOD_LIKE_RECEPTOR_SIGNALING_PATHWAY',\n",
       "  'RIG_I_LIKE_RECEPTOR_SIGNALING_PATHWAY',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY',\n",
       "  'JAK_STAT_SIGNALING_PATHWAY',\n",
       "  'HEMATOPOIETIC_CELL_LINEAGE',\n",
       "  'NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY',\n",
       "  'T_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
       "  'B_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
       "  'FC_EPSILON_RI_SIGNALING_PATHWAY',\n",
       "  'FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS',\n",
       "  'LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION',\n",
       "  'INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION',\n",
       "  'CIRCADIAN_RHYTHM_MAMMAL',\n",
       "  'LONG_TERM_POTENTIATION',\n",
       "  'NEUROTROPHIN_SIGNALING_PATHWAY',\n",
       "  'LONG_TERM_DEPRESSION',\n",
       "  'TASTE_TRANSDUCTION',\n",
       "  'REGULATION_OF_ACTIN_CYTOSKELETON',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY',\n",
       "  'TYPE_II_DIABETES_MELLITUS',\n",
       "  'TYPE_I_DIABETES_MELLITUS',\n",
       "  'MATURITY_ONSET_DIABETES_OF_THE_YOUNG',\n",
       "  'ALDOSTERONE_REGULATED_SODIUM_REABSORPTION',\n",
       "  'VASOPRESSIN_REGULATED_WATER_REABSORPTION',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'AMYOTROPHIC_LATERAL_SCLEROSIS_ALS',\n",
       "  'HUNTINGTONS_DISEASE',\n",
       "  'PRION_DISEASES',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION',\n",
       "  'PATHOGENIC_ESCHERICHIA_COLI_INFECTION',\n",
       "  'LEISHMANIA_INFECTION',\n",
       "  'PATHWAYS_IN_CANCER',\n",
       "  'COLORECTAL_CANCER',\n",
       "  'RENAL_CELL_CARCINOMA',\n",
       "  'PANCREATIC_CANCER',\n",
       "  'ENDOMETRIAL_CANCER',\n",
       "  'GLIOMA',\n",
       "  'PROSTATE_CANCER',\n",
       "  'THYROID_CANCER',\n",
       "  'BASAL_CELL_CARCINOMA',\n",
       "  'MELANOMA',\n",
       "  'BLADDER_CANCER',\n",
       "  'CHRONIC_MYELOID_LEUKEMIA',\n",
       "  'ACUTE_MYELOID_LEUKEMIA',\n",
       "  'SMALL_CELL_LUNG_CANCER',\n",
       "  'NON_SMALL_CELL_LUNG_CANCER',\n",
       "  'ASTHMA',\n",
       "  'AUTOIMMUNE_THYROID_DISEASE',\n",
       "  'SYSTEMIC_LUPUS_ERYTHEMATOSUS',\n",
       "  'ALLOGRAFT_REJECTION',\n",
       "  'GRAFT_VERSUS_HOST_DISEASE',\n",
       "  'PRIMARY_IMMUNODEFICIENCY',\n",
       "  'HYPERTROPHIC_CARDIOMYOPATHY_HCM',\n",
       "  'ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC',\n",
       "  'DILATED_CARDIOMYOPATHY',\n",
       "  'VIRAL_MYOCARDITIS'],\n",
       " 'IDH3B': ['CITRATE_CYCLE_TCA_CYCLE'],\n",
       " 'DLST': ['CITRATE_CYCLE_TCA_CYCLE', 'LYSINE_DEGRADATION'],\n",
       " 'CS': ['CITRATE_CYCLE_TCA_CYCLE', 'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'LOC642502': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'LOC283398': ['CITRATE_CYCLE_TCA_CYCLE', 'PROPANOATE_METABOLISM'],\n",
       " 'FH': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PATHWAYS_IN_CANCER',\n",
       "  'RENAL_CELL_CARCINOMA'],\n",
       " 'SDHD': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'OGDH': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'SDHB': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'IDH3A': ['CITRATE_CYCLE_TCA_CYCLE'],\n",
       " 'SDHC': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'IDH2': ['CITRATE_CYCLE_TCA_CYCLE', 'GLUTATHIONE_METABOLISM', 'PEROXISOME'],\n",
       " 'IDH1': ['CITRATE_CYCLE_TCA_CYCLE', 'GLUTATHIONE_METABOLISM', 'PEROXISOME'],\n",
       " 'ACO1': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'ACLY': ['CITRATE_CYCLE_TCA_CYCLE'],\n",
       " 'MDH2': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'MDH1': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'OGDHL': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'PC': ['CITRATE_CYCLE_TCA_CYCLE', 'PYRUVATE_METABOLISM'],\n",
       " 'SDHA': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'SUCLG1': ['CITRATE_CYCLE_TCA_CYCLE', 'PROPANOATE_METABOLISM'],\n",
       " 'SUCLA2': ['CITRATE_CYCLE_TCA_CYCLE', 'PROPANOATE_METABOLISM'],\n",
       " 'SUCLG2': ['CITRATE_CYCLE_TCA_CYCLE', 'PROPANOATE_METABOLISM'],\n",
       " 'IDH3G': ['CITRATE_CYCLE_TCA_CYCLE'],\n",
       " 'ACO2': ['CITRATE_CYCLE_TCA_CYCLE',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'RPE': ['PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS'],\n",
       " 'RPIA': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'PGLS': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'PRPS2': ['PENTOSE_PHOSPHATE_PATHWAY', 'PURINE_METABOLISM'],\n",
       " 'TALDO1': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'TKT': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'TKTL2': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'PGD': ['PENTOSE_PHOSPHATE_PATHWAY', 'GLUTATHIONE_METABOLISM'],\n",
       " 'RBKS': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'H6PD': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'LOC729020': ['PENTOSE_PHOSPHATE_PATHWAY',\n",
       "  'PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS'],\n",
       " 'PRPS1L1': ['PENTOSE_PHOSPHATE_PATHWAY', 'PURINE_METABOLISM'],\n",
       " 'PRPS1': ['PENTOSE_PHOSPHATE_PATHWAY', 'PURINE_METABOLISM'],\n",
       " 'DERA': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'G6PD': ['PENTOSE_PHOSPHATE_PATHWAY', 'GLUTATHIONE_METABOLISM'],\n",
       " 'TKTL1': ['PENTOSE_PHOSPHATE_PATHWAY'],\n",
       " 'UGT1A10': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A8': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A7': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A6': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT2B28': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A5': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'CRYL1': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS'],\n",
       " 'UGDH': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'UGT2A1': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'GUSB': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES',\n",
       "  'LYSOSOME'],\n",
       " 'UGT1A9': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'DCXR': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS'],\n",
       " 'DHDH': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'UGT2B11': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGP2': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'XYLB': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS'],\n",
       " 'UGT2B10': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'AKR1B1': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'GALACTOSE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM'],\n",
       " 'UGT2B7': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT2B4': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT2A3': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A4': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT2B17': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A1': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT1A3': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UGT2B15': ['PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS',\n",
       "  'ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'MPI': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'PMM2': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'PMM1': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GMDS': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'PFKFB4': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'MTMR6': ['FRUCTOSE_AND_MANNOSE_METABOLISM', 'RIBOFLAVIN_METABOLISM'],\n",
       " 'PHPT1': ['FRUCTOSE_AND_MANNOSE_METABOLISM', 'RIBOFLAVIN_METABOLISM'],\n",
       " 'PFKFB3': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'FUK': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'PFKFB2': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'MTMR1': ['FRUCTOSE_AND_MANNOSE_METABOLISM', 'RIBOFLAVIN_METABOLISM'],\n",
       " 'PFKFB1': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'AKR1B10': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'FPGT': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'KHK': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'MTMR2': ['FRUCTOSE_AND_MANNOSE_METABOLISM', 'RIBOFLAVIN_METABOLISM'],\n",
       " 'MTMR7': ['FRUCTOSE_AND_MANNOSE_METABOLISM', 'RIBOFLAVIN_METABOLISM'],\n",
       " 'TSTA3': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'SORD': ['FRUCTOSE_AND_MANNOSE_METABOLISM'],\n",
       " 'GMPPA': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GMPPB': ['FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GALK1': ['GALACTOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GLB1': ['GALACTOSE_METABOLISM',\n",
       "  'OTHER_GLYCAN_DEGRADATION',\n",
       "  'GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'SPHINGOLIPID_METABOLISM',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES',\n",
       "  'LYSOSOME'],\n",
       " 'GALE': ['GALACTOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'B4GALT1': ['GALACTOSE_METABOLISM',\n",
       "  'N_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'LALBA': ['GALACTOSE_METABOLISM'],\n",
       " 'MGAM': ['GALACTOSE_METABOLISM', 'STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'GALT': ['GALACTOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GLA': ['GALACTOSE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'SPHINGOLIPID_METABOLISM',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'LYSOSOME'],\n",
       " 'GANC': ['GALACTOSE_METABOLISM', 'STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'LCT': ['GALACTOSE_METABOLISM'],\n",
       " 'GALK2': ['GALACTOSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'B4GALT2': ['GALACTOSE_METABOLISM',\n",
       "  'N_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'GAA': ['GALACTOSE_METABOLISM', 'STARCH_AND_SUCROSE_METABOLISM', 'LYSOSOME'],\n",
       " 'MIOX': ['ASCORBATE_AND_ALDARATE_METABOLISM',\n",
       "  'INOSITOL_PHOSPHATE_METABOLISM'],\n",
       " 'CPT1A': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'CPT1C': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'ACADS': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'ACADSB': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'ACADL': ['FATTY_ACID_METABOLISM', 'PPAR_SIGNALING_PATHWAY'],\n",
       " 'ACADM': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY'],\n",
       " 'CYP4A11': ['FATTY_ACID_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION'],\n",
       " 'ACAT2': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TERPENOID_BACKBONE_BIOSYNTHESIS'],\n",
       " 'ACADVL': ['FATTY_ACID_METABOLISM'],\n",
       " 'ACAT1': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'PYRUVATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TERPENOID_BACKBONE_BIOSYNTHESIS'],\n",
       " 'ACAA2': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'HADH': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'HADHB': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'HADHA': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS'],\n",
       " 'CYP4A22': ['FATTY_ACID_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION'],\n",
       " 'ACSL6': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'ECHS1': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'ACSL5': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'EHHADH': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'GCDH': ['FATTY_ACID_METABOLISM',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'ACOX3': ['FATTY_ACID_METABOLISM',\n",
       "  'ALPHA_LINOLENIC_ACID_METABOLISM',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'ACSL1': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'ACAA1': ['FATTY_ACID_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'CPT2': ['FATTY_ACID_METABOLISM', 'PPAR_SIGNALING_PATHWAY'],\n",
       " 'CPT1B': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'ACOX1': ['FATTY_ACID_METABOLISM',\n",
       "  'ALPHA_LINOLENIC_ACID_METABOLISM',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'ECI2': ['FATTY_ACID_METABOLISM', 'PEROXISOME'],\n",
       " 'ECI1': ['FATTY_ACID_METABOLISM'],\n",
       " 'ACSL3': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'ACSL4': ['FATTY_ACID_METABOLISM',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'ADIPOCYTOKINE_SIGNALING_PATHWAY'],\n",
       " 'SOAT1': ['STEROID_BIOSYNTHESIS'],\n",
       " 'LSS': ['STEROID_BIOSYNTHESIS'],\n",
       " 'SQLE': ['STEROID_BIOSYNTHESIS'],\n",
       " 'EBP': ['STEROID_BIOSYNTHESIS'],\n",
       " 'CYP51A1': ['STEROID_BIOSYNTHESIS'],\n",
       " 'DHCR7': ['STEROID_BIOSYNTHESIS'],\n",
       " 'CYP27B1': ['STEROID_BIOSYNTHESIS'],\n",
       " 'DHCR24': ['STEROID_BIOSYNTHESIS'],\n",
       " 'HSD17B7': ['STEROID_BIOSYNTHESIS', 'STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'MSMO1': ['STEROID_BIOSYNTHESIS'],\n",
       " 'FDFT1': ['STEROID_BIOSYNTHESIS'],\n",
       " 'SC5DL': ['STEROID_BIOSYNTHESIS'],\n",
       " 'LIPA': ['STEROID_BIOSYNTHESIS', 'LYSOSOME'],\n",
       " 'CEL': ['STEROID_BIOSYNTHESIS', 'GLYCEROLIPID_METABOLISM'],\n",
       " 'TM7SF2': ['STEROID_BIOSYNTHESIS'],\n",
       " 'NSDHL': ['STEROID_BIOSYNTHESIS'],\n",
       " 'SOAT2': ['STEROID_BIOSYNTHESIS'],\n",
       " 'CYP46A1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS'],\n",
       " 'SLC27A5': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'PPAR_SIGNALING_PATHWAY'],\n",
       " 'BAAT': ['PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
       "  'PEROXISOME'],\n",
       " 'CYP7B1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'AKR1C4': ['PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'HSD17B4': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'PEROXISOME'],\n",
       " 'SCP2': ['PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'AKR1D1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'ACOX2': ['PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'PPAR_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME'],\n",
       " 'HSD3B7': ['PRIMARY_BILE_ACID_BIOSYNTHESIS'],\n",
       " 'CYP27A1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'PPAR_SIGNALING_PATHWAY'],\n",
       " 'AMACR': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'PEROXISOME'],\n",
       " 'CYP7A1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS',\n",
       "  'STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'PPAR_SIGNALING_PATHWAY'],\n",
       " 'CYP8B1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS', 'PPAR_SIGNALING_PATHWAY'],\n",
       " 'CYP39A1': ['PRIMARY_BILE_ACID_BIOSYNTHESIS'],\n",
       " 'CH25H': ['PRIMARY_BILE_ACID_BIOSYNTHESIS'],\n",
       " 'SRD5A3': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP3A5': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'HSD3B2': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'HSD3B1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'COMT': ['STEROID_HORMONE_BIOSYNTHESIS', 'TYROSINE_METABOLISM'],\n",
       " 'SULT2B1': ['STEROID_HORMONE_BIOSYNTHESIS', 'SULFUR_METABOLISM'],\n",
       " 'CYP3A4': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'CYP11A1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'AKR1C3': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'ARACHIDONIC_ACID_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'HSD11B2': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'ALDOSTERONE_REGULATED_SODIUM_REABSORPTION'],\n",
       " 'SULT1E1': ['STEROID_HORMONE_BIOSYNTHESIS', 'SULFUR_METABOLISM'],\n",
       " 'HSD11B1': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'ALDOSTERONE_REGULATED_SODIUM_REABSORPTION'],\n",
       " 'HSD17B2': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'HSD17B3': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP21A2': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'HSD17B1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP19A1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP17A1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP11B2': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP11B1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'HSD17B6': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP3A7': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'HSD17B8': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'AKR1C2': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'AKR1C1': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'CYP1A1': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'CYP1B1': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450'],\n",
       " 'STS': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'SRD5A2': ['STEROID_HORMONE_BIOSYNTHESIS', 'PROSTATE_CANCER'],\n",
       " 'SRD5A1': ['STEROID_HORMONE_BIOSYNTHESIS'],\n",
       " 'CYP3A43': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'HSD17B12': ['STEROID_HORMONE_BIOSYNTHESIS',\n",
       "  'BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS'],\n",
       " 'ATP6V1G1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'UQCR10': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA5': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA4': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX6CP3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'PPA2': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP5J2': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'NDUFS7': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'CYTB': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V0A1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V1G2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V0B': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP5O': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6AP1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX8C': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX5B': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX4I1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V0A4': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP12A': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP6': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP8': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5H': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'LOC100133737': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5G1P5': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCRQ': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1H': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP5D': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'LHPP': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP6V0D2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX7A2L': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5G3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1C2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP5G2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5G1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5F1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5E': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX5A': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'TCIRG1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'UQCRHL': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP4B': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP6V1D': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP4A': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP5A1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5J': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5I': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP6V1A': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V1E1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'NDUFA4L2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1B2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V1B1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V1C1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ATP6V0C': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX10': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM'],\n",
       " 'COX8A': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX7C': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1G3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX7B': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX7A1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX7A2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX6C': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX17': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP5B': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX6B1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP5C1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'LOC644310': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1F': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX7B2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V0A2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'UQCRB': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V0D1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'LYSOSOME',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'COX6A2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX6A1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCRH': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCRFS1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCRC2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCRC1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA10': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFAB1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V0E2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ND6': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'NDUFA9': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ND5': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'NDUFA7': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA8': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'UQCR11': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFA6': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFV3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ATP6V1E2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'ND4': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'ND4L': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'LOC727947': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ND2': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'COX6B2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'PPA1': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ND3': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'COX4I2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ND1': ['OXIDATIVE_PHOSPHORYLATION', 'PARKINSONS_DISEASE'],\n",
       " 'ATP5L': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'ATP6V0E1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION'],\n",
       " 'NDUFA11': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'NDUFB10': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS8': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFC1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFC2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFV2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB6': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS4': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB7': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFV1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB8': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS6': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB9': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS5': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS2': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB4': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFS3': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NDUFB5': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'CYC1': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'CARDIAC_MUSCLE_CONTRACTION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'PARKINSONS_DISEASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'COX11': ['OXIDATIVE_PHOSPHORYLATION'],\n",
       " 'COX15': ['OXIDATIVE_PHOSPHORYLATION',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM'],\n",
       " 'POLR2G': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'NT5C2': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'POLR2H': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ENPP3': ['PURINE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RIBOFLAVIN_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS'],\n",
       " 'POLR2E': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'POLR2F': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ENPP1': ['PURINE_METABOLISM',\n",
       "  'STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'RIBOFLAVIN_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS'],\n",
       " 'XDH': ['PURINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES', 'PEROXISOME'],\n",
       " 'POLR2I': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'POLR2J': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'POLE3': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR'],\n",
       " 'ADSS': ['PURINE_METABOLISM', 'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'ADSL': ['PURINE_METABOLISM', 'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'NME6': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'POLR1D': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'PNPT1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'RNA_DEGRADATION'],\n",
       " 'POLR2K': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'PDE11A': ['PURINE_METABOLISM'],\n",
       " 'POLD3': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR',\n",
       "  'MISMATCH_REPAIR',\n",
       "  'HOMOLOGOUS_RECOMBINATION'],\n",
       " 'POLR2L': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'AK1': ['PURINE_METABOLISM'],\n",
       " 'NME1-NME2': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'AK2': ['PURINE_METABOLISM'],\n",
       " 'AK4': ['PURINE_METABOLISM'],\n",
       " 'POLA2': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'DNA_REPLICATION'],\n",
       " 'CANT1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'AMPD1': ['PURINE_METABOLISM'],\n",
       " 'AMPD3': ['PURINE_METABOLISM'],\n",
       " 'AMPD2': ['PURINE_METABOLISM'],\n",
       " 'NME4': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'GUCY2D': ['PURINE_METABOLISM', 'OLFACTORY_TRANSDUCTION'],\n",
       " 'NME1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'ATIC': ['PURINE_METABOLISM', 'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'NME2': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'DGUOK': ['PURINE_METABOLISM'],\n",
       " 'POLR3C': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'NME3': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'POLR3G': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'POLR3F': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'PRUNE': ['PURINE_METABOLISM'],\n",
       " 'C17orf48': ['PURINE_METABOLISM', 'GLYCEROPHOSPHOLIPID_METABOLISM'],\n",
       " 'PDE4B': ['PURINE_METABOLISM'],\n",
       " 'PDE4C': ['PURINE_METABOLISM'],\n",
       " 'PDE3B': ['PURINE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION'],\n",
       " 'PDE4A': ['PURINE_METABOLISM'],\n",
       " 'NUDT2': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'GMPR2': ['PURINE_METABOLISM'],\n",
       " 'PDE6C': ['PURINE_METABOLISM'],\n",
       " 'PDE6D': ['PURINE_METABOLISM'],\n",
       " 'PDE4D': ['PURINE_METABOLISM'],\n",
       " 'PDE6A': ['PURINE_METABOLISM'],\n",
       " 'PRIM1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'DNA_REPLICATION'],\n",
       " 'PRIM2': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'DNA_REPLICATION'],\n",
       " 'AK5': ['PURINE_METABOLISM'],\n",
       " 'GMPR': ['PURINE_METABOLISM'],\n",
       " 'NME5': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'PDE7B': ['PURINE_METABOLISM'],\n",
       " 'PDE5A': ['PURINE_METABOLISM'],\n",
       " 'POLD4': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR',\n",
       "  'MISMATCH_REPAIR',\n",
       "  'HOMOLOGOUS_RECOMBINATION'],\n",
       " 'GUCY1A2': ['PURINE_METABOLISM',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'LONG_TERM_DEPRESSION'],\n",
       " 'FHIT': ['PURINE_METABOLISM',\n",
       "  'SMALL_CELL_LUNG_CANCER',\n",
       "  'NON_SMALL_CELL_LUNG_CANCER'],\n",
       " 'ADCY3': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'OLFACTORY_TRANSDUCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'VASOPRESSIN_REGULATED_WATER_REABSORPTION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'ADCY2': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'ADCY1': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'LONG_TERM_POTENTIATION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'POLA1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'DNA_REPLICATION'],\n",
       " 'PDE3A': ['PURINE_METABOLISM',\n",
       "  'INSULIN_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION'],\n",
       " 'PDE2A': ['PURINE_METABOLISM'],\n",
       " 'PDE1C': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'OLFACTORY_TRANSDUCTION'],\n",
       " 'ADSSL1': ['PURINE_METABOLISM', 'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'ADA': ['PURINE_METABOLISM', 'PRIMARY_IMMUNODEFICIENCY'],\n",
       " 'ENTPD8': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'POLE2': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR'],\n",
       " 'PDE8A': ['PURINE_METABOLISM'],\n",
       " 'PDE9A': ['PURINE_METABOLISM'],\n",
       " 'PDE1B': ['PURINE_METABOLISM', 'CALCIUM_SIGNALING_PATHWAY'],\n",
       " 'NUDT9': ['PURINE_METABOLISM'],\n",
       " 'POLD1': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR',\n",
       "  'MISMATCH_REPAIR',\n",
       "  'HOMOLOGOUS_RECOMBINATION'],\n",
       " 'GDA': ['PURINE_METABOLISM'],\n",
       " 'POLD2': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR',\n",
       "  'MISMATCH_REPAIR',\n",
       "  'HOMOLOGOUS_RECOMBINATION'],\n",
       " 'POLE': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR'],\n",
       " 'PDE6B': ['PURINE_METABOLISM'],\n",
       " 'PDE7A': ['PURINE_METABOLISM'],\n",
       " 'POLR1E': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'RNA_POLYMERASE'],\n",
       " 'PAICS': ['PURINE_METABOLISM'],\n",
       " 'NT5C1A': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'POLR2B': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'POLR3A': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'PDE6H': ['PURINE_METABOLISM'],\n",
       " 'POLR2A': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'PDE6G': ['PURINE_METABOLISM'],\n",
       " 'POLR2D': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'DCK': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'POLR2C': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'ADCY8': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'LONG_TERM_POTENTIATION',\n",
       "  'TASTE_TRANSDUCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'ADCY9': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'VASOPRESSIN_REGULATED_WATER_REABSORPTION',\n",
       "  'VIBRIO_CHOLERAE_INFECTION',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'ADCY6': ['PURINE_METABOLISM',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'TASTE_TRANSDUCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'VASOPRESSIN_REGULATED_WATER_REABSORPTION',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'AK7': ['PURINE_METABOLISM'],\n",
       " 'ADCY7': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'ADCY5': ['PURINE_METABOLISM',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'GMPS': ['PURINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'ADCY10': ['PURINE_METABOLISM'],\n",
       " 'POLR3K': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'POLR1B': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'RNA_POLYMERASE'],\n",
       " 'PNP': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'RRM2B': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'P53_SIGNALING_PATHWAY'],\n",
       " 'PDE1A': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'TASTE_TRANSDUCTION'],\n",
       " 'POLR2J2': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'HPRT1': ['PURINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'GART': ['PURINE_METABOLISM', 'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'PAPSS1': ['PURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'SULFUR_METABOLISM'],\n",
       " 'PAPSS2': ['PURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'SULFUR_METABOLISM'],\n",
       " 'NT5M': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'ADCY4': ['PURINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'CHEMOKINE_SIGNALING_PATHWAY',\n",
       "  'OOCYTE_MEIOSIS',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'TASTE_TRANSDUCTION',\n",
       "  'GNRH_SIGNALING_PATHWAY',\n",
       "  'PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
       "  'MELANOGENESIS',\n",
       "  'DILATED_CARDIOMYOPATHY'],\n",
       " 'IMPDH1': ['PURINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'IMPDH2': ['PURINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'POLR3H': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'POLR2J3': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'POLR1A': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'RNA_POLYMERASE'],\n",
       " 'POLR3B': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'GUCY1B3': ['PURINE_METABOLISM',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'LONG_TERM_DEPRESSION'],\n",
       " 'GUCY2C': ['PURINE_METABOLISM'],\n",
       " 'ENTPD4': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'LYSOSOME'],\n",
       " 'GUCY1A3': ['PURINE_METABOLISM',\n",
       "  'VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
       "  'GAP_JUNCTION',\n",
       "  'LONG_TERM_DEPRESSION'],\n",
       " 'GUK1': ['PURINE_METABOLISM'],\n",
       " 'GUCY2F': ['PURINE_METABOLISM'],\n",
       " 'ADK': ['PURINE_METABOLISM'],\n",
       " 'ITPA': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'POLR1C': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'PRHOXNB': ['PURINE_METABOLISM'],\n",
       " 'POLE4': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'DNA_REPLICATION',\n",
       "  'BASE_EXCISION_REPAIR',\n",
       "  'NUCLEOTIDE_EXCISION_REPAIR'],\n",
       " 'RRM1': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM'],\n",
       " 'POLR3D': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'RRM2': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'P53_SIGNALING_PATHWAY'],\n",
       " 'NT5C3': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'PDE8B': ['PURINE_METABOLISM'],\n",
       " 'POLR3GL': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'RNA_POLYMERASE',\n",
       "  'CYTOSOLIC_DNA_SENSING_PATHWAY'],\n",
       " 'NT5E': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'NPR2': ['PURINE_METABOLISM', 'VASCULAR_SMOOTH_MUSCLE_CONTRACTION'],\n",
       " 'NPR1': ['PURINE_METABOLISM', 'VASCULAR_SMOOTH_MUSCLE_CONTRACTION'],\n",
       " 'PDE10A': ['PURINE_METABOLISM'],\n",
       " 'ENTPD6': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'ENTPD2': ['PURINE_METABOLISM'],\n",
       " 'ENTPD5': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'ENTPD3': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'NUDT5': ['PURINE_METABOLISM'],\n",
       " 'ENTPD1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'PFAS': ['PURINE_METABOLISM'],\n",
       " 'NT5C1B': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'APRT': ['PURINE_METABOLISM'],\n",
       " 'PPAT': ['PURINE_METABOLISM', 'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'NME7': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM'],\n",
       " 'NT5C': ['PURINE_METABOLISM',\n",
       "  'PYRIMIDINE_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM'],\n",
       " 'ALLC': ['PURINE_METABOLISM'],\n",
       " 'ZNRD1': ['PURINE_METABOLISM', 'PYRIMIDINE_METABOLISM', 'RNA_POLYMERASE'],\n",
       " 'TYMS': ['PYRIMIDINE_METABOLISM', 'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'TXNRD1': ['PYRIMIDINE_METABOLISM'],\n",
       " 'DHODH': ['PYRIMIDINE_METABOLISM'],\n",
       " 'UPRT': ['PYRIMIDINE_METABOLISM'],\n",
       " 'DTYMK': ['PYRIMIDINE_METABOLISM'],\n",
       " 'CTPS': ['PYRIMIDINE_METABOLISM'],\n",
       " 'CAD': ['PYRIMIDINE_METABOLISM',\n",
       "  'ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'CMPK2': ['PYRIMIDINE_METABOLISM'],\n",
       " 'DCTD': ['PYRIMIDINE_METABOLISM'],\n",
       " 'UCK1': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UPB1': ['PYRIMIDINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'TXNRD2': ['PYRIMIDINE_METABOLISM'],\n",
       " 'CMPK1': ['PYRIMIDINE_METABOLISM'],\n",
       " 'UPP2': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'CTPS2': ['PYRIMIDINE_METABOLISM'],\n",
       " 'CDA': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'DUT': ['PYRIMIDINE_METABOLISM'],\n",
       " 'UCKL1': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'DPYS': ['PYRIMIDINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'DPYD': ['PYRIMIDINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'TYMP': ['PYRIMIDINE_METABOLISM',\n",
       "  'DRUG_METABOLISM_OTHER_ENZYMES',\n",
       "  'BLADDER_CANCER'],\n",
       " 'UMPS': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UCK2': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'TK1': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'TK2': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'UPP1': ['PYRIMIDINE_METABOLISM', 'DRUG_METABOLISM_OTHER_ENZYMES'],\n",
       " 'AK3': ['PYRIMIDINE_METABOLISM'],\n",
       " 'GLUD2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'GFPT2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'AGXT2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'CPS1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM'],\n",
       " 'GLS2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'ABAT': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'GLUD1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'GLS': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM',\n",
       "  'PROXIMAL_TUBULE_BICARBONATE_RECLAMATION'],\n",
       " 'ASL': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'IL4I1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'GPT': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'GFPT1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GPT2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'DDO': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM', 'PEROXISOME'],\n",
       " 'ACY3': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM'],\n",
       " 'GAD1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TYPE_I_DIABETES_MELLITUS'],\n",
       " 'GAD2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TYPE_I_DIABETES_MELLITUS'],\n",
       " 'GOT2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM'],\n",
       " 'AGXT': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'PEROXISOME'],\n",
       " 'GLUL': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'NITROGEN_METABOLISM'],\n",
       " 'NIT2': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM'],\n",
       " 'GOT1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM'],\n",
       " 'ALDH4A1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ALDH5A1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'ASS1': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ASNS': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM', 'NITROGEN_METABOLISM'],\n",
       " 'ASPA': ['ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM'],\n",
       " 'ALAS1': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM'],\n",
       " 'ALAS2': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'PORPHYRIN_AND_CHLOROPHYLL_METABOLISM'],\n",
       " 'GLYCTK': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'GLYCEROLIPID_METABOLISM',\n",
       "  'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'MAOB': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'MAOA': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'AOC2': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM'],\n",
       " 'GATM': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'SDS': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'AOC3': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM'],\n",
       " 'GNMT': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'SHMT1': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'GCAT': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'PHGDH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'DMGDH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'SHMT2': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'SRR': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'PSAT1': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'BHMT': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'GLDC': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'GAMT': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PSPH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'SARDH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'AMT': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ONE_CARBON_POOL_BY_FOLATE',\n",
       "  'NITROGEN_METABOLISM'],\n",
       " 'CHDH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM'],\n",
       " 'CBS': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'CTH': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'NITROGEN_METABOLISM'],\n",
       " 'PIPOX': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'LYSINE_DEGRADATION',\n",
       "  'PEROXISOME'],\n",
       " 'DAO': ['GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'PEROXISOME'],\n",
       " 'AMD1': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'SRM': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM'],\n",
       " 'ADI1': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'AHCY': ['CYSTEINE_AND_METHIONINE_METABOLISM', 'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'DNMT1': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'TRDMT1': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'DNMT3A': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'DNMT3B': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'SMS': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM'],\n",
       " 'ENOPH1': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'CDO1': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'TAURINE_AND_HYPOTAURINE_METABOLISM'],\n",
       " 'APIP': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'TAT': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM'],\n",
       " 'AHCYL1': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MTAP': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'MAT2B': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MTR': ['CYSTEINE_AND_METHIONINE_METABOLISM', 'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'MAT1A': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'AHCYL2': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MAT2A': ['CYSTEINE_AND_METHIONINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MPST': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'DNMT3L': ['CYSTEINE_AND_METHIONINE_METABOLISM'],\n",
       " 'AOX1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM',\n",
       "  'NICOTINATE_AND_NICOTINAMIDE_METABOLISM',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'HSD17B10': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'ALZHEIMERS_DISEASE'],\n",
       " 'OXCT1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'MCEE': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'DBT': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'ALDH6A1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'INOSITOL_PHOSPHATE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'HMGCL': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'PEROXISOME'],\n",
       " 'IVD': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'BCKDHB': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'ACAD8': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'PCCB': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'HIBCH': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BETA_ALANINE_METABOLISM',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'HMGCS2': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TERPENOID_BACKBONE_BIOSYNTHESIS',\n",
       "  'PPAR_SIGNALING_PATHWAY'],\n",
       " 'HMGCS1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM',\n",
       "  'TERPENOID_BACKBONE_BIOSYNTHESIS'],\n",
       " 'PCCA': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'PROPANOATE_METABOLISM'],\n",
       " 'HIBADH': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'OXCT2': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'BUTANOATE_METABOLISM'],\n",
       " 'MUT': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION', 'PROPANOATE_METABOLISM'],\n",
       " 'MCCC2': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'BCAT1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS'],\n",
       " 'AUH': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'BCAT2': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
       "  'VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'PANTOTHENATE_AND_COA_BIOSYNTHESIS'],\n",
       " 'BCKDHA': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'MCCC1': ['VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION'],\n",
       " 'VARS2': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'IARS2': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'LARS': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'LARS2': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'VARS': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'IARS': ['VALINE_LEUCINE_AND_ISOLEUCINE_BIOSYNTHESIS',\n",
       "  'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'SUV39H2': ['LYSINE_DEGRADATION'],\n",
       " 'AASDHPPT': ['LYSINE_DEGRADATION'],\n",
       " 'AASDH': ['LYSINE_DEGRADATION'],\n",
       " 'PLOD2': ['LYSINE_DEGRADATION'],\n",
       " 'PLOD1': ['LYSINE_DEGRADATION'],\n",
       " 'TMLHE': ['LYSINE_DEGRADATION'],\n",
       " 'EHMT2': ['LYSINE_DEGRADATION'],\n",
       " 'SETD8': ['LYSINE_DEGRADATION'],\n",
       " 'SETMAR': ['LYSINE_DEGRADATION'],\n",
       " 'BBOX1': ['LYSINE_DEGRADATION'],\n",
       " 'AASS': ['LYSINE_DEGRADATION'],\n",
       " 'SUV420H1': ['LYSINE_DEGRADATION'],\n",
       " 'SUV39H1': ['LYSINE_DEGRADATION'],\n",
       " 'ASH1L': ['LYSINE_DEGRADATION', 'TIGHT_JUNCTION'],\n",
       " 'SETD1B': ['LYSINE_DEGRADATION'],\n",
       " 'DOT1L': ['LYSINE_DEGRADATION'],\n",
       " 'WHSC1': ['LYSINE_DEGRADATION'],\n",
       " 'SETD1A': ['LYSINE_DEGRADATION'],\n",
       " 'SUV420H2': ['LYSINE_DEGRADATION'],\n",
       " 'SETD7': ['LYSINE_DEGRADATION'],\n",
       " 'WHSC1L1': ['LYSINE_DEGRADATION'],\n",
       " 'EHMT1': ['LYSINE_DEGRADATION'],\n",
       " 'SETDB1': ['LYSINE_DEGRADATION'],\n",
       " 'AADAT': ['LYSINE_DEGRADATION', 'TRYPTOPHAN_METABOLISM'],\n",
       " 'SETD2': ['LYSINE_DEGRADATION'],\n",
       " 'SETDB2': ['LYSINE_DEGRADATION'],\n",
       " 'PLOD3': ['LYSINE_DEGRADATION'],\n",
       " 'NSD1': ['LYSINE_DEGRADATION'],\n",
       " 'ADC': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ARG1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ARG2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'CKMT1A': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'P4HA2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'CKB': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'NOS3': ['ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'VEGF_SIGNALING_PATHWAY'],\n",
       " 'NOS2': ['ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'PEROXISOME',\n",
       "  'LEISHMANIA_INFECTION',\n",
       "  'PATHWAYS_IN_CANCER',\n",
       "  'SMALL_CELL_LUNG_CANCER'],\n",
       " 'NOS1': ['ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'CALCIUM_SIGNALING_PATHWAY',\n",
       "  'LONG_TERM_DEPRESSION',\n",
       "  'ALZHEIMERS_DISEASE',\n",
       "  'AMYOTROPHIC_LATERAL_SCLEROSIS_ALS'],\n",
       " 'P4HA1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ABP1': ['ARGININE_AND_PROLINE_METABOLISM',\n",
       "  'HISTIDINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'LAP3': ['ARGININE_AND_PROLINE_METABOLISM', 'GLUTATHIONE_METABOLISM'],\n",
       " 'AGMAT': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PYCR1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ALDH18A1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'SAT1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'NAGS': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'P4HA3': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'OTC': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PRODH2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'OAT': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'CKM': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ACY1': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'SAT2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'CKMT1B': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PYCRL': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PRODH': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'CKMT2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'PYCR2': ['ARGININE_AND_PROLINE_METABOLISM'],\n",
       " 'ODC1': ['ARGININE_AND_PROLINE_METABOLISM', 'GLUTATHIONE_METABOLISM'],\n",
       " 'CNDP1': ['HISTIDINE_METABOLISM', 'BETA_ALANINE_METABOLISM'],\n",
       " 'METTL6': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'WBSCR22': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'HAL': ['HISTIDINE_METABOLISM', 'NITROGEN_METABOLISM'],\n",
       " 'HNMT': ['HISTIDINE_METABOLISM'],\n",
       " 'DDC': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'TRYPTOPHAN_METABOLISM'],\n",
       " 'METTL2B': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'HEMK1': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'LCMT2': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'UROC1': ['HISTIDINE_METABOLISM'],\n",
       " 'TRMT11': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'FTCD': ['HISTIDINE_METABOLISM', 'ONE_CARBON_POOL_BY_FOLATE'],\n",
       " 'AMDHD1': ['HISTIDINE_METABOLISM'],\n",
       " 'LCMT1': ['HISTIDINE_METABOLISM',\n",
       "  'TYROSINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM'],\n",
       " 'HDC': ['HISTIDINE_METABOLISM'],\n",
       " 'HPD': ['TYROSINE_METABOLISM', 'PHENYLALANINE_METABOLISM'],\n",
       " 'TYR': ['TYROSINE_METABOLISM', 'RIBOFLAVIN_METABOLISM', 'MELANOGENESIS'],\n",
       " 'HGD': ['TYROSINE_METABOLISM'],\n",
       " 'GSTZ1': ['TYROSINE_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'DBH': ['TYROSINE_METABOLISM'],\n",
       " 'TYRP1': ['TYROSINE_METABOLISM', 'MELANOGENESIS'],\n",
       " 'MIF': ['TYROSINE_METABOLISM', 'PHENYLALANINE_METABOLISM'],\n",
       " 'NAT6': ['TYROSINE_METABOLISM',\n",
       "  'PHENYLALANINE_METABOLISM',\n",
       "  'LIMONENE_AND_PINENE_DEGRADATION'],\n",
       " 'PNMT': ['TYROSINE_METABOLISM'],\n",
       " 'DCT': ['TYROSINE_METABOLISM', 'MELANOGENESIS'],\n",
       " 'TPO': ['TYROSINE_METABOLISM',\n",
       "  'CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
       "  'JAK_STAT_SIGNALING_PATHWAY',\n",
       "  'HEMATOPOIETIC_CELL_LINEAGE',\n",
       "  'AUTOIMMUNE_THYROID_DISEASE'],\n",
       " 'TH': ['TYROSINE_METABOLISM', 'PARKINSONS_DISEASE'],\n",
       " 'FAH': ['TYROSINE_METABOLISM'],\n",
       " 'PAH': ['PHENYLALANINE_METABOLISM'],\n",
       " 'PRDX6': ['PHENYLALANINE_METABOLISM'],\n",
       " 'IDO2': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'AANAT': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'WARS': ['TRYPTOPHAN_METABOLISM', 'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'IDO1': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'CAT': ['TRYPTOPHAN_METABOLISM',\n",
       "  'PEROXISOME',\n",
       "  'AMYOTROPHIC_LATERAL_SCLEROSIS_ALS'],\n",
       " 'TPH1': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'AFMID': ['TRYPTOPHAN_METABOLISM', 'GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM'],\n",
       " 'CYP1A2': ['TRYPTOPHAN_METABOLISM',\n",
       "  'LINOLEIC_ACID_METABOLISM',\n",
       "  'RETINOL_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'ASMT': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'KMO': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'WARS2': ['TRYPTOPHAN_METABOLISM', 'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'INMT': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'TDO2': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'HAAO': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'KYNU': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'TPH2': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'ACMSD': ['TRYPTOPHAN_METABOLISM'],\n",
       " 'MLYCD': ['BETA_ALANINE_METABOLISM', 'PROPANOATE_METABOLISM', 'PEROXISOME'],\n",
       " 'GGT7': ['TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'CSAD': ['TAURINE_AND_HYPOTAURINE_METABOLISM'],\n",
       " 'GGT5': ['TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GGT1': ['TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GGT6': ['TAURINE_AND_HYPOTAURINE_METABOLISM',\n",
       "  'SELENOAMINO_ACID_METABOLISM',\n",
       "  'GLUTATHIONE_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'ADO': ['TAURINE_AND_HYPOTAURINE_METABOLISM'],\n",
       " 'SEPHS1': ['SELENOAMINO_ACID_METABOLISM'],\n",
       " 'SEPHS2': ['SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MARS2': ['SELENOAMINO_ACID_METABOLISM', 'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'SCLY': ['SELENOAMINO_ACID_METABOLISM'],\n",
       " 'MARS': ['SELENOAMINO_ACID_METABOLISM', 'AMINOACYL_TRNA_BIOSYNTHESIS'],\n",
       " 'GSTP1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'PATHWAYS_IN_CANCER',\n",
       "  'PROSTATE_CANCER'],\n",
       " 'GSTT2': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTT1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTO1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTA5': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'MGST2': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'MGST1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'MGST3': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTA3': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTM1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GPX7': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GSTA4': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GPX6': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GSTM4': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GGCT': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GSTM3': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTM2': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTM5': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTA1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSTA2': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'GSR': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GSS': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GCLC': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GSTK1': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450',\n",
       "  'PEROXISOME'],\n",
       " 'GPX5': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'TXNDC12': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GPX1': ['GLUTATHIONE_METABOLISM',\n",
       "  'ARACHIDONIC_ACID_METABOLISM',\n",
       "  'AMYOTROPHIC_LATERAL_SCLEROSIS_ALS',\n",
       "  'HUNTINGTONS_DISEASE'],\n",
       " 'GPX2': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GPX3': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'GPX4': ['GLUTATHIONE_METABOLISM', 'ARACHIDONIC_ACID_METABOLISM'],\n",
       " 'OPLAH': ['GLUTATHIONE_METABOLISM'],\n",
       " 'GCLM': ['GLUTATHIONE_METABOLISM'],\n",
       " 'ANPEP': ['GLUTATHIONE_METABOLISM',\n",
       "  'RENIN_ANGIOTENSIN_SYSTEM',\n",
       "  'HEMATOPOIETIC_CELL_LINEAGE'],\n",
       " 'GSTO2': ['GLUTATHIONE_METABOLISM',\n",
       "  'METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
       "  'DRUG_METABOLISM_CYTOCHROME_P450'],\n",
       " 'AMY2B': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'UXS1': ['STARCH_AND_SUCROSE_METABOLISM',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GYS1': ['STARCH_AND_SUCROSE_METABOLISM', 'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'GYS2': ['STARCH_AND_SUCROSE_METABOLISM', 'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'GBA3': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'TREH': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'AMY1A': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'AMY1B': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'AMY1C': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'AMY2A': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'PYGB': ['STARCH_AND_SUCROSE_METABOLISM', 'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'PYGM': ['STARCH_AND_SUCROSE_METABOLISM', 'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'PYGL': ['STARCH_AND_SUCROSE_METABOLISM', 'INSULIN_SIGNALING_PATHWAY'],\n",
       " 'PGM2L1': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'AGL': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'SI': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'GBE1': ['STARCH_AND_SUCROSE_METABOLISM'],\n",
       " 'ALG13': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DOLPP1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'RPN1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG14': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN1B1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG3': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT5': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'RPN2': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'STT3A': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT3': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DAD1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT2': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG12': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'TUSC3': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN1C1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DPM2': ['N_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSYLPHOSPHATIDYLINOSITOL_GPI_ANCHOR_BIOSYNTHESIS'],\n",
       " 'DPM1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GANAB': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT4A': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG10B': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'STT3B': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN1A2': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG10': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG11': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG8': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG2': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DPAGT1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'RFT1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DPM3': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'DDOST': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT4B': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG6': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN2A2': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN1A1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MAN2A1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ST6GAL1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'B4GALT3': ['N_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'ALG5': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MGAT5B': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ALG9': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'MOGS': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'FUT8': ['N_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " 'MGAT1': ['N_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ENGASE': ['OTHER_GLYCAN_DEGRADATION'],\n",
       " 'MANBA': ['OTHER_GLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'MAN2B1': ['OTHER_GLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'GBA': ['OTHER_GLYCAN_DEGRADATION', 'SPHINGOLIPID_METABOLISM', 'LYSOSOME'],\n",
       " 'NEU4': ['OTHER_GLYCAN_DEGRADATION', 'SPHINGOLIPID_METABOLISM'],\n",
       " 'NEU2': ['OTHER_GLYCAN_DEGRADATION', 'SPHINGOLIPID_METABOLISM'],\n",
       " 'NEU1': ['OTHER_GLYCAN_DEGRADATION', 'SPHINGOLIPID_METABOLISM', 'LYSOSOME'],\n",
       " 'FUCA1': ['OTHER_GLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'FUCA2': ['OTHER_GLYCAN_DEGRADATION'],\n",
       " 'AGA': ['OTHER_GLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'MAN2C1': ['OTHER_GLYCAN_DEGRADATION'],\n",
       " 'MAN2B2': ['OTHER_GLYCAN_DEGRADATION'],\n",
       " 'NEU3': ['OTHER_GLYCAN_DEGRADATION', 'SPHINGOLIPID_METABOLISM'],\n",
       " 'HEXB': ['OTHER_GLYCAN_DEGRADATION',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES',\n",
       "  'LYSOSOME'],\n",
       " 'HEXA': ['OTHER_GLYCAN_DEGRADATION',\n",
       "  'AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
       "  'GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES',\n",
       "  'LYSOSOME'],\n",
       " 'GALNT4': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNTL2': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNTL5': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT6': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT5': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNTL1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNTL6': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT13': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GCNT3': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT10': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ST6GALNAC1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT9': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT7': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GCNT4': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT11': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GCNT1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT8': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT14': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'B3GNT6': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'C1GALT1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNTL4': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT12': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT2': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT3': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'WBSCR17': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'GALNT1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'ST3GAL2': ['O_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES'],\n",
       " 'ST3GAL1': ['O_GLYCAN_BIOSYNTHESIS',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES'],\n",
       " 'B4GALT5': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'C1GALT1C1': ['O_GLYCAN_BIOSYNTHESIS'],\n",
       " 'AMDHD2': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GNPDA2': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GNPNAT1': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'NANS': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'CYB5R3': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'NANP': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'CYB5R1': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GNE': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'NAGK': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'CHIT1': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'UAP1': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'RENBP': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'CHIA': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'NPL': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'CMAS': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'PGM3': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'GNPDA1': ['AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM'],\n",
       " 'HS3ST3A1': ['GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'HPSE': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'HPSE2': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'HYAL3': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'GNS': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'HYAL4': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'HYAL1': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'HYAL2': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'SPAM1': ['GLYCOSAMINOGLYCAN_DEGRADATION'],\n",
       " 'HGSNAT': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'GALNS': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'IDUA': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'HS3ST3B1': ['GLYCOSAMINOGLYCAN_DEGRADATION',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'SGSH': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'NAGLU': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'IDS': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'ARSB': ['GLYCOSAMINOGLYCAN_DEGRADATION', 'LYSOSOME'],\n",
       " 'DSE': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'B3GALT6': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHPF': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'XYLT1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHSY3': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'XYLT2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHST14': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'B3GAT1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHSY1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'CHST11': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'SULFUR_METABOLISM'],\n",
       " 'CHPF2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'CHST15': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'B4GALT7': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHST7': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'UST': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'CHST13': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'SULFUR_METABOLISM'],\n",
       " 'CSGALNACT2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'B3GAT2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CSGALNACT1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'CHST3': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE'],\n",
       " 'CHST12': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'SULFUR_METABOLISM'],\n",
       " 'B3GAT3': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
       "  'GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE'],\n",
       " 'CHST2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " 'B3GNT1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'B3GNT2': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'CHST1': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " 'B3GNT7': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " 'B4GALT4': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'CHST4': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " 'ST3GAL3': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE',\n",
       "  'GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES'],\n",
       " 'CHST6': ['GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE'],\n",
       " ...}"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathways_pulito_simbolo={}  \n",
    "for riga in pathways_array:\n",
    "    pathway = riga[0]\n",
    "    lista_geni_pathway= list(riga[1:])\n",
    "    for simbolo in lista_geni_pathway:\n",
    "        lis=[]\n",
    "        if simbolo not in pathways_pulito_simbolo:\n",
    "            lis.append(pathway)\n",
    "            pathways_pulito_simbolo[simbolo]= lis\n",
    "        elif simbolo in pathways_pulito_simbolo and pathway not in pathways_pulito_simbolo[simbolo]:\n",
    "            pathways_pulito_simbolo[simbolo].append(pathway)\n",
    "pathways_pulito_simbolo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(pathways_pulito_simbolo)\n",
    "simboli_pathways_no_interat=[]\n",
    "for simbolo in pathways_pulito_simbolo:\n",
    "    if simbolo not in idgene_simbolo.values():\n",
    "        simboli_pathways_no_interat.append(simbolo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "859"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(simboli_pathways_no_interat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "ufficiale_simbolo_id= simbolo_idgene   \n",
    "simboli_solo_in_pathways=[]            \n",
    "for simb in pathways_pulito_simbolo:\n",
    "    if simb not in ufficiale_simbolo_id:\n",
    "        ufficiale_simbolo_id[simb]=''  \n",
    "        simboli_solo_in_pathways.append(simb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating complete Interactome dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11115"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interattoma_completo= diz\n",
    "len(diz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13458"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i in range (0, 141272):\n",
    "    gene_uscita= interattoma.loc[i,'ID gene uscita']\n",
    "    gene_entrata= interattoma.loc[i,'ID gene entrata']\n",
    "    l=[]\n",
    "    if gene_entrata not in interattoma_completo:\n",
    "        l.append(gene_uscita)\n",
    "        interattoma_completo[gene_entrata]= l\n",
    "    else:\n",
    "        if gene_uscita not in interattoma_completo[gene_entrata]:\n",
    "            interattoma_completo[gene_entrata].append(gene_uscita)\n",
    "len(interattoma_completo) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Faccio ufficiale_simbolo_id, arricchendo con nuovi simboli che compaiono nei pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11115"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diz={}  \n",
    "for i in range (0, 141272):\n",
    "    gene_i= interattoma.loc[i,'ID gene uscita']\n",
    "    valore= interattoma.loc[i,'ID gene entrata']\n",
    "    l=[]\n",
    "    if gene_i not in diz:\n",
    "        l.append(valore)\n",
    "        diz[gene_i]= l\n",
    "    else:\n",
    "        diz[gene_i].append(valore)\n",
    "len(diz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idgene_simbolo.pop(4663)  #lo tolgo perchè non ha un simbolo, è stato cambiato con 7504 che già esiste separatamente..l'ho sgamato dai vari errori che mi dava python.."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <center> Choosing the disease to analyze"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Diseases</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 590</th>\n",
       "      <th>Unnamed: 591</th>\n",
       "      <th>Unnamed: 592</th>\n",
       "      <th>Unnamed: 593</th>\n",
       "      <th>Unnamed: 594</th>\n",
       "      <th>Unnamed: 595</th>\n",
       "      <th>Unnamed: 596</th>\n",
       "      <th>Unnamed: 597</th>\n",
       "      <th>Unnamed: 598</th>\n",
       "      <th>Unnamed: 599</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>carbohydrate metabolism inborn errors</td>\n",
       "      <td>374291</td>\n",
       "      <td>4669</td>\n",
       "      <td>56052</td>\n",
       "      <td>6389</td>\n",
       "      <td>189</td>\n",
       "      <td>2998</td>\n",
       "      <td>9382</td>\n",
       "      <td>4125</td>\n",
       "      <td>2592</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Diseases   Genes  Unnamed: 2  Unnamed: 3  \\\n",
       "18  carbohydrate metabolism inborn errors  374291        4669       56052   \n",
       "\n",
       "    Unnamed: 4  Unnamed: 5  Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  \\\n",
       "18        6389         189        2998        9382        4125        2592   \n",
       "\n",
       "        ...       Unnamed: 590  Unnamed: 591  Unnamed: 592  Unnamed: 593  \\\n",
       "18      ...                NaN           NaN           NaN           NaN   \n",
       "\n",
       "    Unnamed: 594  Unnamed: 595  Unnamed: 596  Unnamed: 597  Unnamed: 598  \\\n",
       "18           NaN           NaN           NaN           NaN           NaN   \n",
       "\n",
       "    Unnamed: 599  \n",
       "18           NaN  \n",
       "\n",
       "[1 rows x 600 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disease_name= 'carbohydrate metabolism inborn errors'\n",
    "\n",
    "ulcer = seeds[seeds['Diseases']== disease_name]\n",
    "ulcer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "78\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Diseases      carbohydrate metabolism inborn errors\n",
       "Genes                                        374291\n",
       "Unnamed: 2                                     4669\n",
       "Unnamed: 3                                    56052\n",
       "Unnamed: 4                                     6389\n",
       "Name: 18, dtype: object"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ulcer = seeds.iloc[18].dropna()\n",
    "print(len(ulcer))\n",
    "ulcer.head() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "77"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geni_ulcer= []   \n",
    "for gene in ulcer:\n",
    "    if gene != 'carbohydrate metabolism inborn errors':\n",
    "        geni_ulcer.append(gene)\n",
    "len(geni_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13457"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(idgene_simbolo) #13457 giusto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "dizio_count={}\n",
    "annotazioni_ulcer={}  \n",
    "annotaz_ulcer = diz_annotazioni_da_lista(annotazioni_ulcer, geni_ulcer, dizio_count, 'count annotazioni in ulcer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtro_interattoma_ulcer=[gene for gene in idgene_simbolo if gene not in geni_ulcer]  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13380"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(filtro_interattoma_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz_vuoto={}\n",
    "dizio_count={}\n",
    "media_ann1000_ulcer= media_tab_random (tabella_random_id(diz_vuoto, dizio_count, 56, 1000, filtro_interattoma_ulcer),1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count annotazioni in ulcer</th>\n",
       "      <th>media count annotazioni su 1000 campioni random</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43312</th>\n",
       "      <td>15</td>\n",
       "      <td>2.079853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5980</th>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5977</th>\n",
       "      <td>8</td>\n",
       "      <td>1.016949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6027</th>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6486</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5978</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6094</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6488</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6091</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19388</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       count annotazioni in ulcer  \\\n",
       "43312                          15   \n",
       "5980                           11   \n",
       "5977                            8   \n",
       "6027                            8   \n",
       "6486                            6   \n",
       "5978                            6   \n",
       "6094                            6   \n",
       "6488                            6   \n",
       "6091                            6   \n",
       "19388                           5   \n",
       "\n",
       "       media count annotazioni su 1000 campioni random  \n",
       "43312                                         2.079853  \n",
       "5980                                               NaN  \n",
       "5977                                          1.016949  \n",
       "6027                                               NaN  \n",
       "6486                                               NaN  \n",
       "5978                                               NaN  \n",
       "6094                                               NaN  \n",
       "6488                                               NaN  \n",
       "6091                                               NaN  \n",
       "19388                                              NaN  "
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "confronto_10ulcer_random= annotaz_ulcer.join(media_ann1000_ulcer).head(10)\n",
    "confronto_10ulcer_random"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "confronto_10random_ulcer= media_ann1000_ulcer.join(annotaz_ulcer).head(10)\n",
    "top_20_ulcer= lista_n_annotazioni(media_ann1000_ulcer, annotaz_ulcer,10,10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_15_ulcer= list(pd.Series(top_20_ulcer).value_counts().index)\n",
    "len(togli_annot_ridondanti(top_15_ulcer))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "temp= []\n",
    "geni_con_top_ann_ulcer = geni_con_top_annotazioni(temp ,top_15_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "77"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simboli_geni_ulcer=[]\n",
    "for idgene in geni_ulcer:\n",
    "    if idgene_simbolo[idgene] not in simboli_geni_ulcer: \n",
    "        simboli_geni_ulcer.append(idgene_simbolo[idgene])\n",
    "len(simboli_geni_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5201"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtro_pathways_ulcer =[]\n",
    "for simbolo in pathways_pulito_simbolo:\n",
    "        if simbolo not in simboli_geni_ulcer and simbolo not in filtro_pathways_ulcer:\n",
    "            filtro_pathways_ulcer.append(simbolo)\n",
    "len(filtro_pathways_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "77"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_simboli_ulcer={}\n",
    "for geneid in geni_ulcer:\n",
    "    if geneid in idgene_simbolo:\n",
    "        id_simboli_ulcer[geneid]= idgene_simbolo[geneid]\n",
    "len(id_simboli_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz_pathways_ulcer= {}\n",
    "for riga in pathways_array:  \n",
    "    for simbolo in riga:    \n",
    "        if simbolo in id_simboli_ulcer.values(): \n",
    "            if simbolo not in diz_pathways_ulcer:  \n",
    "                diz_pathways_ulcer[simbolo]= [riga[0]]\n",
    "            if simbolo in diz_pathways_ulcer and riga[0] not in diz_pathways_ulcer[simbolo]:\n",
    "                diz_pathways_ulcer[simbolo].append(riga[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ulcer34_con_pat = list(diz_pathways_ulcer.keys()) \n",
    "len(ulcer34_con_pat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_pathways_ulcer={}\n",
    "for lista in diz_pathways_ulcer.values():\n",
    "    for pathway in lista:\n",
    "        if pathway not in count_pathways_ulcer:\n",
    "            count_pathways_ulcer[pathway]=1\n",
    "        else:\n",
    "            count_pathways_ulcer[pathway]+=1\n",
    "count_pathways_ulcer= tabella_da_diz(count_pathways_ulcer)\n",
    "count_pathways_ulcer.columns= ['Count annotaz pathways nei geni ulcer']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "diz_vuoto={}\n",
    "dizio_count={}\n",
    "path_1000_34= tabella_random_pathways (diz_vuoto, dizio_count, 34, 1000, filtro_pathways_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "media_path_1000_34= media_tab_random_pathways(path_1000_34, 1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_6_pathways_ulcer= lista_n_annotazioni(media_path_1000_34, count_pathways_ulcer, 3,3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_6_pathways_ulcer= list(pd.Series(top_6_pathways_ulcer).unique())\n",
    "len(top_6_pathways_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1245"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l=[]\n",
    "geni_top_6_pathw_ulcer= geni_con_top_pathways(l,top_6_pathways_ulcer)\n",
    "geni_top_6_pathw_ulcer= list(pd.Series(geni_top_6_pathw_ulcer).dropna())\n",
    "len(geni_top_6_pathw_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['LYSOSOME',\n",
       " 'N_GLYCAN_BIOSYNTHESIS',\n",
       " 'INSULIN_SIGNALING_PATHWAY',\n",
       " 'OLFACTORY_TRANSDUCTION',\n",
       " 'PATHWAYS_IN_CANCER',\n",
       " 'NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION']"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"Queste erano quando avevo lasciato un crohn nell analisi di alz--> ['CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
    " 'JAK_STAT_SIGNALING_PATHWAY',\n",
    " 'PATHWAYS_IN_CANCER',\n",
    " 'OLFACTORY_TRANSDUCTION']\"\"\"\n",
    "\n",
    "top_6_pathways_ulcer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>6355</th>\n",
       "      <th>6366</th>\n",
       "      <th>45944</th>\n",
       "      <th>6027</th>\n",
       "      <th>6091</th>\n",
       "      <th>6094</th>\n",
       "      <th>6351</th>\n",
       "      <th>43312</th>\n",
       "      <th>43066</th>\n",
       "      <th>7186</th>\n",
       "      <th>19388</th>\n",
       "      <th>6357</th>\n",
       "      <th>6486</th>\n",
       "      <th>122</th>\n",
       "      <th>6488</th>\n",
       "      <th>5977</th>\n",
       "      <th>5978</th>\n",
       "      <th>5980</th>\n",
       "      <th>7165</th>\n",
       "      <th>6915</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      6355   6366   45944  6027   6091   6094   6351   43312  43066  7186   \\\n",
       "1         0      0      0      0      0      0      0      1      0      0   \n",
       "1026      0      0      0      0      0      0      0      0      1      0   \n",
       "\n",
       "      19388  6357   6486   122    6488   5977   5978   5980   7165   6915   \n",
       "1         0      0      0      0      0      0      0      0      0      0  \n",
       "1026      0      1      0      0      0      0      0      0      0      0  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "incidenza_ann_ulcer={}\n",
    "incidenza_ann_ulcer= dizio_vettori_incidenza (incidenza_ann_ulcer, geni_con_top_ann_ulcer,top_15_ulcer )\n",
    "incidenza_ann_ulcer.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LYSOSOME</th>\n",
       "      <th>N_GLYCAN_BIOSYNTHESIS</th>\n",
       "      <th>INSULIN_SIGNALING_PATHWAY</th>\n",
       "      <th>OLFACTORY_TRANSDUCTION</th>\n",
       "      <th>PATHWAYS_IN_CANCER</th>\n",
       "      <th>NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCK</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBP1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HK2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HK1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HK3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      LYSOSOME  N_GLYCAN_BIOSYNTHESIS  INSULIN_SIGNALING_PATHWAY  \\\n",
       "GCK          0                      0                          1   \n",
       "FBP1         0                      0                          1   \n",
       "HK2          0                      0                          1   \n",
       "HK1          0                      0                          1   \n",
       "HK3          0                      0                          1   \n",
       "\n",
       "      OLFACTORY_TRANSDUCTION  PATHWAYS_IN_CANCER  \\\n",
       "GCK                        0                   0   \n",
       "FBP1                       0                   0   \n",
       "HK2                        0                   0   \n",
       "HK1                        0                   0   \n",
       "HK3                        0                   0   \n",
       "\n",
       "      NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION  \n",
       "GCK                                         0  \n",
       "FBP1                                        0  \n",
       "HK2                                         0  \n",
       "HK1                                         0  \n",
       "HK3                                         0  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d={}\n",
    "incidenza_6pat_ulcer= dizio_vettori_incidenza_pathways(d, geni_top_6_pathw_ulcer, top_6_pathways_ulcer )\n",
    "incidenza_6pat_ulcer.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "495\n"
     ]
    }
   ],
   "source": [
    "count_putative_ulcer={}\n",
    "count_putative_seeds_ulcer= {}\n",
    "count_putative_ulcer['analisi fatte']=0\n",
    "count_putative_seeds_ulcer['analisi fatte']=0\n",
    "\n",
    "\n",
    "firstneig_913ulcer= []\n",
    "for nodo in geni_ulcer:\n",
    "    for adiacente in interattoma_completo[nodo]:\n",
    "        if adiacente not in firstneig_913ulcer:\n",
    "            firstneig_913ulcer.append(adiacente)\n",
    "print(len(firstneig_913ulcer))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1> <center> Annotation Driven Clustering Method </center> </h1> \n",
    "<h3> Phase 1: from generation on N (1000) random samples, to the all clustering processes performed both based on annotations and on pathways </h3> <br>\n",
    "\n",
    "<h3> Phase 2: For each analysis, it is chosen the optimal number of clusters as the one having more seed genes of the disease, with the constraint of keeping at least 90% of the initial (that is currently known) genes of the disease. \n",
    "Again, this is done both according to annotations and pathways; the two optimal clusters are then intersected to obtain a set of putative seed genes or, in other words, the new genes predicted to be seeds. </h3> "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Phase 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "#os.chdir('C:\\\\Users\\\\feder\\\\jupyter\\\\tesi network medicine')\n",
    "#os.mkdir('colite ulcerosa')\n",
    "os.chdir('C:\\\\Users\\\\feder\\\\jupyter\\\\tesi network medicine\\\\colite ulcerosa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "analisi n°91\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 40 seed su 4465 cioè il 0.9 %\n",
      "None\n",
      "N= 2\n",
      "33 su 3242 e i rossi tot  495 , il 15.3 % 0\n",
      "7 su 1223 e i rossi tot  129 , il 10.5 % 1\n",
      "None\n",
      "N= 3\n",
      "6 su 797 e i rossi tot  145 , il 18.2 % 0\n",
      "27 su 2465 e i rossi tot  357 , il 14.5 % 1\n",
      "7 su 1203 e i rossi tot  122 , il 10.1 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 18 seed su 1215 cioè il 1.5 %\n",
      "None\n",
      "N= 2\n",
      "18 su 826 e i rossi tot  152 , il 18.4 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\feder\\AppData\\Local\\Continuum\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py:22: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 su 561 e i rossi tot  137 , il 24.4 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "2 su 265 e i rossi tot  15 , il 5.7 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°92\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  7165\n",
      "annot ridondante:  45893\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 39 seed su 4863 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "7 su 1249 e i rossi tot  133 , il 10.6 % 0\n",
      "32 su 3614 e i rossi tot  509 , il 14.1 % 1\n",
      "None\n",
      "N= 3\n",
      "26 su 2740 e i rossi tot  386 , il 14.1 % 0\n",
      "7 su 1208 e i rossi tot  127 , il 10.5 % 1\n",
      "6 su 915 e i rossi tot  129 , il 14.1 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "annot ridondante:  CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION\n",
      "geni della malattia: \n",
      "ci sono 19 seed su 808 cioè il 2.4 %\n",
      "None\n",
      "N= 2\n",
      "5 su 525 e i rossi tot  127 , il 24.2 % 0\n",
      "14 su 283 e i rossi tot  75 , il 26.5 % 1\n",
      "None\n",
      "N= 3\n",
      "3 su 325 e i rossi tot  90 , il 27.7 % 0\n",
      "11 su 263 e i rossi tot  66 , il 25.1 % 1\n",
      "5 su 220 e i rossi tot  46 , il 20.9 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°93\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  7186\n",
      "annot ridondante:  45893\n",
      "geni della malattia: \n",
      "ci sono 36 seed su 4303 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "29 su 3091 e i rossi tot  475 , il 15.4 % 0\n",
      "7 su 1212 e i rossi tot  129 , il 10.6 % 1\n",
      "None\n",
      "N= 3\n",
      "7 su 1201 e i rossi tot  120 , il 10.0 % 0\n",
      "23 su 2289 e i rossi tot  339 , il 14.8 % 1\n",
      "6 su 813 e i rossi tot  145 , il 17.8 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "annot ridondante:  MAPK_SIGNALING_PATHWAY\n",
      "geni della malattia: \n",
      "ci sono 19 seed su 1256 cioè il 1.5 %\n",
      "None\n",
      "N= 2\n",
      "19 su 867 e i rossi tot  214 , il 24.7 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "16 su 382 e i rossi tot  104 , il 27.2 % 0\n",
      "3 su 485 e i rossi tot  110 , il 22.7 % 1\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°94\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  7165\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  7186\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 36 seed su 4411 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "7 su 1214 e i rossi tot  128 , il 10.5 % 0\n",
      "29 su 3197 e i rossi tot  478 , il 15.0 % 1\n",
      "None\n",
      "N= 3\n",
      "23 su 2393 e i rossi tot  339 , il 14.2 % 0\n",
      "6 su 816 e i rossi tot  146 , il 17.9 % 1\n",
      "7 su 1202 e i rossi tot  121 , il 10.1 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 22 seed su 1411 cioè il 1.6 %\n",
      "None\n",
      "N= 2\n",
      "22 su 1022 e i rossi tot  201 , il 19.7 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "19 su 694 e i rossi tot  110 , il 15.9 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "3 su 328 e i rossi tot  91 , il 27.7 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°95\n",
      "annot ridondante:  7165\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  7186\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 38 seed su 4566 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "7 su 1220 e i rossi tot  128 , il 10.5 % 0\n",
      "31 su 3346 e i rossi tot  498 , il 14.9 % 1\n",
      "None\n",
      "N= 3\n",
      "7 su 1202 e i rossi tot  121 , il 10.1 % 0\n",
      "25 su 2551 e i rossi tot  359 , il 14.1 % 1\n",
      "6 su 813 e i rossi tot  146 , il 18.0 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 20 seed su 1392 cioè il 1.4 %\n",
      "None\n",
      "N= 2\n",
      "20 su 1003 e i rossi tot  189 , il 18.8 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "17 su 675 e i rossi tot  98 , il 14.5 % 0\n",
      "3 su 328 e i rossi tot  91 , il 27.7 % 1\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°96\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  6915\n",
      "annot ridondante:  45892\n",
      "geni della malattia: \n",
      "ci sono 37 seed su 4161 cioè il 0.9 %\n",
      "None\n",
      "N= 2\n",
      "30 su 2953 e i rossi tot  496 , il 16.8 % 0\n",
      "7 su 1208 e i rossi tot  124 , il 10.3 % 1\n",
      "None\n",
      "N= 3\n",
      "24 su 2156 e i rossi tot  349 , il 16.2 % 0\n",
      "7 su 1198 e i rossi tot  121 , il 10.1 % 1\n",
      "6 su 807 e i rossi tot  150 , il 18.6 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 20 seed su 1290 cioè il 1.6 %\n",
      "None\n",
      "N= 2\n",
      "20 su 901 e i rossi tot  192 , il 21.3 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "18 su 629 e i rossi tot  174 , il 27.7 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "2 su 272 e i rossi tot  18 , il 6.6 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°97\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  7165\n",
      "annot ridondante:  7186\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 36 seed su 4441 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "29 su 3222 e i rossi tot  487 , il 15.1 % 0\n",
      "7 su 1219 e i rossi tot  129 , il 10.6 % 1\n",
      "None\n",
      "N= 3\n",
      "23 su 2422 e i rossi tot  349 , il 14.4 % 0\n",
      "7 su 1203 e i rossi tot  121 , il 10.1 % 1\n",
      "6 su 816 e i rossi tot  146 , il 17.9 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "annot ridondante:  CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION\n",
      "geni della malattia: \n",
      "ci sono 18 seed su 1120 cioè il 1.6 %\n",
      "None\n",
      "N= 2\n",
      "18 su 731 e i rossi tot  151 , il 20.7 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "5 su 470 e i rossi tot  78 , il 16.6 % 0\n",
      "0 su 384 e i rossi tot  1 , il 0.3 % 1\n",
      "13 su 266 e i rossi tot  73 , il 27.4 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°98\n",
      "annot ridondante:  7165\n",
      "annot ridondante:  45944\n",
      "geni della malattia: \n",
      "ci sono 39 seed su 4519 cioè il 0.9 %\n",
      "None\n",
      "N= 2\n",
      "32 su 3300 e i rossi tot  504 , il 15.3 % 0\n",
      "7 su 1219 e i rossi tot  129 , il 10.6 % 1\n",
      "None\n",
      "N= 3\n",
      "26 su 2501 e i rossi tot  365 , il 14.6 % 0\n",
      "6 su 819 e i rossi tot  147 , il 17.9 % 1\n",
      "7 su 1199 e i rossi tot  121 , il 10.1 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 19 seed su 1192 cioè il 1.6 %\n",
      "None\n",
      "N= 2\n",
      "19 su 803 e i rossi tot  202 , il 25.2 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "5 su 520 e i rossi tot  127 , il 24.4 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "14 su 283 e i rossi tot  75 , il 26.5 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°99\n",
      "annot ridondante:  6351\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  45893\n",
      "geni della malattia: \n",
      "ci sono 38 seed su 4593 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "26 su 2691 e i rossi tot  369 , il 13.7 % 0\n",
      "12 su 1902 e i rossi tot  253 , il 13.3 % 1\n",
      "None\n",
      "N= 3\n",
      "25 su 2585 e i rossi tot  355 , il 13.7 % 0\n",
      "7 su 1200 e i rossi tot  121 , il 10.1 % 1\n",
      "6 su 808 e i rossi tot  146 , il 18.1 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "geni della malattia: \n",
      "ci sono 20 seed su 1290 cioè il 1.6 %\n",
      "None\n",
      "N= 2\n",
      "20 su 901 e i rossi tot  192 , il 21.3 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "18 su 629 e i rossi tot  174 , il 27.7 % 0\n",
      "2 su 272 e i rossi tot  18 , il 6.6 % 1\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "analisi n°100\n",
      "annot ridondante:  45944\n",
      "annot ridondante:  122\n",
      "annot ridondante:  45893\n",
      "annot ridondante:  6915\n",
      "geni della malattia: \n",
      "ci sono 38 seed su 4818 cioè il 0.8 %\n",
      "None\n",
      "N= 2\n",
      "31 su 3590 e i rossi tot  556 , il 15.5 % 0\n",
      "7 su 1228 e i rossi tot  128 , il 10.4 % 1\n",
      "None\n",
      "N= 3\n",
      "6 su 930 e i rossi tot  129 , il 13.9 % 0\n",
      "7 su 1180 e i rossi tot  122 , il 10.3 % 1\n",
      "25 su 2708 e i rossi tot  433 , il 16.0 % 2\n",
      "None\n",
      "Analisi sui Pathways\n",
      "annot ridondante:  CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION\n",
      "geni della malattia: \n",
      "ci sono 18 seed su 1174 cioè il 1.5 %\n",
      "None\n",
      "N= 2\n",
      "18 su 785 e i rossi tot  173 , il 22.0 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "None\n",
      "N= 3\n",
      "5 su 525 e i rossi tot  101 , il 19.2 % 0\n",
      "0 su 389 e i rossi tot  1 , il 0.3 % 1\n",
      "13 su 260 e i rossi tot  72 , il 27.7 % 2\n",
      "None\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "min_n_clusters= 2\n",
    "max_n_clusters= 6\n",
    "\n",
    "for j in range(1,101): #I decide how many analysis to make\n",
    "    print('analisi n°'+str(j))\n",
    "    writer = pd.ExcelWriter('CV'+ str(j) +' 20 ann 6 path ulcer.xlsx')\n",
    "    random={}\n",
    "    dizio_count={}\n",
    "    cv_count_ann=tabella_da_diz(count_campione_geni(campione_simboli_random(random, 40, simboli_geni_ulcer),dizio_count))\n",
    "    cv_count_ann.columns=['count annot in CV7']\n",
    "    #pd.DataFrame(list(random.keys())).to_excel(writer,'max 50 simboli ulcer') \n",
    "    #cv7_count_ann.to_excel(writer,'conta di ciascuna annotazione')  \n",
    "    random_sample={}\n",
    "    count_funzioni_random={}\n",
    "    cv_top20_ann= togli_annot_ridondanti(lista_n_annotazioni(\n",
    "    media_tab_random(\n",
    "    tabella_random_id(random_sample, count_funzioni_random, 40, 1000, filtro_interattoma_ulcer), 1000), cv_count_ann, 10, 10))\n",
    "    \n",
    "    cv_top20_ann_def= []\n",
    "    for ann in cv_top20_ann:\n",
    "        if ann not in cv_top20_ann_def:\n",
    "            cv_top20_ann_def.append(ann)\n",
    "    len(cv_top20_ann_def)\n",
    "    #pd.DataFrame(cv_top20_ann_def).to_excel(writer,'max 20 annotaz top')\n",
    "    \n",
    "    cv_simboli_tot=[]\n",
    "    cv_id_tot=[]\n",
    "    cv_simboli_tot= simboli_con_top_annotazioni(cv_simboli_tot, cv_top20_ann)\n",
    "    pd.DataFrame(cv_simboli_tot).to_excel(writer,'simboli top')\n",
    "    for simb in cv_simboli_tot:\n",
    "        if simbolo_idgene[simb] in idgene_simbolo:\n",
    "            cv_id_tot.append(simbolo_idgene[simb])\n",
    "    #pd.DataFrame(cv_id_tot).to_excel(writer,'geni id top')\n",
    "    print('geni della malattia: ')\n",
    "    print(conta_id_in_lista(cv_id_tot, geni_ulcer))\n",
    "    \n",
    "    cv_incidenza={}\n",
    "    cv_incidenza= dizio_vettori_incidenza(cv_incidenza, cv_id_tot, cv_top20_ann)\n",
    "    #cv_incidenza.to_excel(writer, 'mat incidenza')\n",
    "    #print(cv11_incidenza.head(2))\n",
    "    for n in range (min_n_clusters, max_n_clusters):  #decido quanti clusters fare\n",
    "        print('N=',n)\n",
    "        cv_kmeans= KMeans(n_clusters= n, init='k-means++').fit(cv_incidenza)\n",
    "        cv_cluster_n= pd.DataFrame()\n",
    "        cv_cluster_n['idgene'] = cv_incidenza.index.values\n",
    "        cv_cluster_n['cluster'] = cv_kmeans.labels_\n",
    "        if n<=3:\n",
    "            print(uff_resoconto_seeds_cluster(cv_cluster_n, 4, 0, geni_ulcer))\n",
    "        else:\n",
    "            print(uff_resoconto_seeds_cluster(cv_cluster_n, 10, 1, geni_ulcer))\n",
    "        cv_cluster_n.to_excel(writer, 'cluster n'+str(n))\n",
    "        \n",
    "    print('Analisi sui Pathways') #finita l'analisi i-esima per le annotazioni, faccio la i-esima per i pathways\n",
    "    random={}\n",
    "    dizio_count={}\n",
    "    cv_6pat_count_ann= tabella_random_pathways(random, dizio_count, 24, 1, ulcer34_con_pat)\n",
    "    random_sample={}\n",
    "    count_funzioni_random={}\n",
    "    cv_top6_pat= togli_annot_ridondanti(\n",
    "    lista_n_annotazioni(\n",
    "    media_tab_random_pathways(\n",
    "    tabella_random_pathways(random_sample, count_funzioni_random, 24, 1000, filtro_pathways_ulcer), 1000), cv_6pat_count_ann, 3,3))\n",
    "    \n",
    "    cv_top6_pat_def= []\n",
    "    for pat in cv_top6_pat:\n",
    "        if pat not in cv_top6_pat_def:\n",
    "            cv_top6_pat_def.append(pat)\n",
    "    #pd.DataFrame(cv_top6_pat_def).to_excel(writer,'max 6 pathways top')\n",
    "\n",
    "    cv_6pat_simboli_tot=[]\n",
    "    cv_6pat_simboli_tot= togli_nan_da_lista_simboli(\n",
    "        togli_annot_ridondanti(geni_con_top_pathways(cv_6pat_simboli_tot, cv_top6_pat)))\n",
    "    pd.DataFrame(cv_6pat_simboli_tot).to_excel(writer,'simb top')\n",
    "    \n",
    "    print('geni della malattia: ')\n",
    "    print(conta_simboli_in_lista(cv_6pat_simboli_tot, geni_ulcer))\n",
    "    cv_6pat_incidenza={}\n",
    "    cv_6pat_incidenza= dizio_vettori_incidenza_pathways(cv_6pat_incidenza, cv_6pat_simboli_tot, cv_top6_pat)\n",
    "    #cv_6pat_incidenza.to_excel(writer, 'mat incidenza')\n",
    "\n",
    "    for n in range (2,4): #decido quanti clusters fare per ogni analisi\n",
    "        print('N=',n)\n",
    "        cv_6pat_kmeans= KMeans(n_clusters= n, init='k-means++').fit(cv_6pat_incidenza)\n",
    "        cv_6pat_cluster= pd.DataFrame()\n",
    "        cv_6pat_cluster['simbolo gene'] = cv_6pat_incidenza.index.values\n",
    "        cv_6pat_cluster['cluster'] = cv_6pat_kmeans.labels_\n",
    "        cv_6pat_cluster.to_excel(writer, 'cluspat n'+str(n))\n",
    "        if n<=3:\n",
    "            print(resoconto_seeds_cluster_pat(cv_6pat_cluster, 4, 0, geni_ulcer))\n",
    "        else:\n",
    "            print(resoconto_seeds_cluster_pat(cv_6pat_cluster, 10, 0, geni_ulcer))\n",
    "    \n",
    "    #SCELTA DEL CLUSTER OTTIMO la faccio fuori, perchè si può fare solo una volta che viene chiuso il file excel.\n",
    "    writer.save()\n",
    "    writer.close()\n",
    "    print('\\n')    \n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phase 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ANALISI NUMERO:  91\n",
      "il best rapporto è cambiato a:  1.0 e il best cluster è:  2 ,best gruppo: 0\n",
      "\n",
      "\n",
      "tot seeds sono:  40\n",
      "il best cluster è:  2\n",
      "ora per pathways\n",
      "tot seeds sono:  18\n",
      "il best rapporto è cambiato a:  2.2 e il best cluster è:  2 ,best gruppo: 0\n",
      "il best rapporto è cambiato a:  2.9 e il best cluster è:  3 ,best gruppo: 0\n",
      "\n",
      "\n",
      "il best cluster è:  3\n",
      "gruppo annot:  3242 , gruppo path:  561 \n",
      "tot geni comuni 374 ,in %  66.66666666666667 \n",
      "quelli seeds in comune 107 \n",
      "quelli della malattia comuni:  16 lunghezza lista comuni è:  374\n",
      "ci sono 54 candidati sconosciuti che sono: ['RHOA', 'ARHGEF1', 'GNA13', 'ARHGEF12', 'PAK1', 'AKT3', 'IL1RAP', 'PIK3R1', 'IL1A', 'IL1B', 'GRB2', 'CSK', 'PPBP', 'CXCL6', 'CXCL5', 'CXCL1', 'CXCL2', 'JAK3', 'MAPK3', 'PTPN6', 'PTPN11', 'RAF1', 'STAT6', 'VAV1', 'CXCR4', 'STAM', 'SOCS1', 'SOCS3', 'CRLF2', 'CCR2', 'CSF2RB', 'CSF3R', 'EGFR', 'EPOR', 'IL3RA', 'IL4R', 'IL5RA', 'IL6ST', 'TNFRSF25', 'FLT1', 'ITGB1', 'KDR', 'IL10RA', 'IL10RB', 'IFNB1', 'EPO', 'BMPR2', 'EZR', 'IFNA1', 'IFNA14', 'IL23A', 'JAK1', 'IL2RG', 'IL7'] mentre 28 sono candidati ma rossi:  ['IFNGR1', 'IFNGR2', 'TNF', 'AKT1', 'AKT2', 'CREBBP', 'TGFBR1', 'TGFBR2', 'HLA-DRA', 'CXCL3', 'KIT', 'PDGFRB', 'PTK2', 'STAT3', 'STAT5B', 'TNFRSF1A', 'TYK2', 'OSMR', 'CCR5', 'GHR', 'IL2RB', 'LTA', 'TNFSF14', 'TNFSF13', 'TNFSF11', 'EP300', 'IL12RB1', 'IL12A']\n"
     ]
    }
   ],
   "source": [
    "min_n_clusters= 2\n",
    "max_n_clusters= 6\n",
    "\n",
    "for n_analisi in range(91,92):  #For each analysis done (and saved in Excel)\n",
    "    print('ANALISI NUMERO: ', n_analisi)\n",
    "    simboli_top= list(pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'simboli top').squeeze())\n",
    "    tot_seeds_malattia= quanti_seed_simb_tot(simboli_top, geni_ulcer)\n",
    "    k_best=2 \n",
    "    best_rapporto= 0\n",
    "\n",
    "    best_group= pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'cluster n'+ str(k_best)) \n",
    "    for k in range(2,4): \n",
    "        for gruppo in range(0,k):  \n",
    "            group= pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'cluster n'+ str(k)) \n",
    "            group= list(group [group['cluster']== gruppo]['idgene'])\n",
    "            if rapporto_idseeds_gruppo(group, geni_ulcer)> best_rapporto and quanti_seed_totali(\n",
    "                group, geni_ulcer)>= 0.70 * tot_seeds_malattia:\n",
    "                best_rapporto= rapporto_idseeds_gruppo(group, geni_ulcer)\n",
    "                best_group= group\n",
    "                k_best= k\n",
    "                print('il best rapporto è cambiato a: ', best_rapporto, 'e il best cluster è: ', k_best, ',best gruppo:', gruppo)\n",
    "    \n",
    "    print('\\n')\n",
    "    print('tot seeds sono: ', tot_seeds_malattia)\n",
    "    print('il best cluster è: ', k_best)\n",
    "    \n",
    "    print('ora per pathways')\n",
    "    \n",
    "    simb_top_pat= list(pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'simb top').squeeze())\n",
    "    tot_seeds_pat= quanti_seed_simb_tot(simb_top_pat, geni_ulcer)\n",
    "    print('tot seeds sono: ', tot_seeds_pat)\n",
    "    k_best_pat=3 \n",
    "    best_rapporto_pat= 0\n",
    "    best_group_pat= pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'cluspat n'+ str(k_best_pat)) \n",
    "    for kpat in range(2,4): \n",
    "        for gruppo_pat in range(0,kpat):  \n",
    "            group_pat= pd.read_excel('CV' + str(n_analisi)+ ' 20 ann 6 path ulcer.xlsx', sheet_name= 'cluspat n'+ str(kpat)) \n",
    "            group_pat= list(group_pat [group_pat['cluster']== gruppo_pat]['simbolo gene'])\n",
    "            if rapporto_simbseeds_gruppo(group_pat, geni_ulcer)> best_rapporto_pat and quanti_seed_simb_tot(group_pat, geni_ulcer)>= 0.73 * tot_seeds_pat:\n",
    "                best_rapporto_pat= rapporto_simbseeds_gruppo(group_pat, geni_ulcer)\n",
    "                best_rapporto_pat= rapporto_simbseeds_gruppo(group_pat, geni_ulcer)\n",
    "                best_group_pat= group_pat\n",
    "                k_best_pat= kpat\n",
    "                print('il best rapporto è cambiato a: ', best_rapporto_pat, 'e il best cluster è: ', k_best_pat, ',best gruppo:', gruppo_pat)\n",
    "\n",
    "    print('\\n')\n",
    "    print('il best cluster è: ', k_best_pat)\n",
    "    \n",
    "    simb_comuni_cv= []\n",
    "    confronto_13_bests(best_group, best_group_pat, geni_ulcer, simb_comuni_cv)\n",
    "    \n",
    "    cv_simboli_nocarbo=[]\n",
    "    cv_simboli_carbo=[]\n",
    "    cv_idgeni_carbo=[]\n",
    "    for simbolo in simb_comuni_cv:\n",
    "        if simbolo in simboli_geni_ulcer:\n",
    "            cv_simboli_carbo.append(simbolo)\n",
    "            cv_idgeni_carbo.append(simbolo_idgene[simbolo])\n",
    "        else:\n",
    "            cv_simboli_nocarbo.append(simbolo)\n",
    "\n",
    "    CV_13_putative=[]\n",
    "    CV_13_putative_seeds=[]\n",
    "    for idgene in firstneig_913ulcer: \n",
    "        if math.isnan(idgene)== False:\n",
    "            if idgene_simbolo[idgene] in cv_simboli_nocarbo:\n",
    "                if idgene not in lista_seeds_unici:\n",
    "                    CV_13_putative.append(idgene_simbolo[idgene])\n",
    "                else:\n",
    "                    CV_13_putative_seeds.append(idgene_simbolo[idgene])\n",
    "    print('ci sono',len(CV_13_putative),'candidati sconosciuti che sono:',\n",
    "          CV_13_putative, 'mentre', len(CV_13_putative_seeds),'sono candidati ma rossi: ',CV_13_putative_seeds)\n",
    "    \n",
    "    for gene in CV_13_putative_seeds:  \n",
    "        if gene not in count_putative_seeds_ulcer:   \n",
    "            count_putative_seeds_ulcer[gene]=1\n",
    "        else:\n",
    "            count_putative_seeds_ulcer[gene]+=1\n",
    "    for gene in CV_13_putative: \n",
    "        if gene not in count_putative_ulcer:\n",
    "            count_putative_ulcer[gene]=1\n",
    "        else:\n",
    "            count_putative_ulcer[gene]+=1\n",
    "    count_putative_seeds_ulcer['analisi fatte']+=1\n",
    "    count_putative_ulcer['analisi fatte']+=1\n",
    "    \n",
    "    \n",
    "    \"\"\"writer = pd.ExcelWriter('CV' + str(n_analisi)+'intersezioni ulcer.xlsx')\n",
    "    pd.DataFrame(pd.Series(simb_comuni_cv)).to_excel(writer, 'simboli comuni')\n",
    "    pd.DataFrame(cv_simboli_nocarbo).to_excel(writer, 'comuni non ulcer')\n",
    "    pd.DataFrame(CV_13_putative).to_excel(writer, 'putative normali')\n",
    "    pd.DataFrame(CV_13_putative_seeds).to_excel(writer, 'putative rossi')\n",
    "    writer.save()\n",
    "    writer.close()\"\"\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### The whole analysis so far was based on datasets of 2013; with the above clustering process, some putative seed genes have been predicted. \n",
    "### Considering a more recent dataset (Disgenet) we can now have the 'labels' and find out which of them was actually found out to be seed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "put_seeds_ulcer= pd.DataFrame(pd.Series(count_putative_seeds_ulcer).sort_values(ascending= False))\n",
    "simb_25put_seeds_ulcer= list(put_seeds_ulcer[put_seeds_ulcer[0] > 39].iloc[1:].index) #\n",
    "len(simb_25put_seeds_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "put_ulcer = pd.DataFrame(pd.Series(count_putative_ulcer).sort_values(ascending= False))\n",
    "simb_49put_ulcer= list(put_ulcer[put_ulcer[0] > 40].iloc[1:].index)\n",
    "len(simb_49put_ulcer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('C:\\\\Users\\\\feder\\\\jupyter\\\\tesi network medicine')\n",
    "diseasome= pd.read_excel('new diseasome.xlsx')\n",
    "copia_dis= diseasome\n",
    "def lower(stringa):\n",
    "    return stringa.lower()\n",
    "\n",
    "def iniziale_stringa (stringa, parte_iniziale):\n",
    "    if stringa== parte_iniziale + stringa[len(parte_iniziale):] :\n",
    "        return True\n",
    "    \n",
    "def parola_in_stringa (stringa, parola):\n",
    "    if parola in stringa:\n",
    "        return True\n",
    "\n",
    "copia_dis['diseaseName']= copia_dis['diseaseName'].apply(lambda x: lower(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "sottoclassi_ulcer = list(pd.Series(copia_dis[copia_dis['diseaseName'].apply(lambda x: parola_in_stringa(x,'ulcerative colitis'))== True]['diseaseName']).value_counts().index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dajeeeee col gene:  CCR5  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  CCR5  e nome malattia chronic ulcerative colitis\n",
      "dajeeeee col gene:  LTA  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  LTA  e nome malattia chronic ulcerative colitis\n",
      "dajeeeee col gene:  TNFRSF1A  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  TNF  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  TNF  e nome malattia chronic ulcerative colitis\n",
      "dajeeeee col gene:  HLA-DRA  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  PDGFRB  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  TGFBR2  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  NFKB1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  NFKB1  e nome malattia pediatric ulcerative colitis\n",
      "8 seeds putative, su 25\n"
     ]
    }
   ],
   "source": [
    "\"\"\" Vediamo se qualcuno dei 25 seeds risulta provocare una di queste sottoclassi \"\"\"\n",
    "validati_seed_ulcer={}\n",
    "for put_seed in simb_25put_seeds_ulcer:\n",
    "    righe_gene= diseasome[diseasome['geneSymbol']== put_seed]\n",
    "    for malattia in sottoclassi_ulcer:\n",
    "        if len(righe_gene[righe_gene['diseaseName']== malattia])>0:\n",
    "            print('dajeeeee col gene: ', put_seed , ' e nome malattia', malattia)\n",
    "            if put_seed not in validati_seed_ulcer:\n",
    "                validati_seed_ulcer[put_seed]= [malattia]\n",
    "            else:\n",
    "                validati_seed_ulcer[put_seed].append(malattia)\n",
    "print(len(validati_seed_ulcer), 'seeds putative, su',  len(simb_25put_seeds_ulcer))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dajeeeee col gene:  CXCL6  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  CXCR4  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL4R  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  KDR  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  CXCL1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  CXCL5  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  CXCL5  e nome malattia pediatric ulcerative colitis\n",
      "dajeeeee col gene:  IL10RB  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  FLT1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL23A  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL1B  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL1B  e nome malattia chronic ulcerative colitis\n",
      "dajeeeee col gene:  IL1A  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL1A  e nome malattia chronic ulcerative colitis\n",
      "dajeeeee col gene:  IL7  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IFNA1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IL10RA  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  IRAK1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  MYD88  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  SOCS3  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  SOCS3  e nome malattia ulcerative colitis in remission\n",
      "dajeeeee col gene:  STAT6  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  STAT6  e nome malattia pediatric ulcerative colitis\n",
      "dajeeeee col gene:  LEPR  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  SOCS1  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  PTPN11  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  EGFR  e nome malattia ulcerative colitis\n",
      "dajeeeee col gene:  NFKBIA  e nome malattia ulcerative colitis\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(23, 'putative, su', 49)"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "validati_put_ulcer={}\n",
    "for put_seed in simb_49put_ulcer:\n",
    "    righe_gene= diseasome[diseasome['geneSymbol']== put_seed]\n",
    "    for malattia in sottoclassi_ulcer:\n",
    "        if len(righe_gene[righe_gene['diseaseName']== malattia])>0:\n",
    "            print('dajeeeee col gene: ', put_seed , ' e nome malattia', malattia)\n",
    "            if put_seed not in validati_put_ulcer:\n",
    "                validati_put_ulcer[put_seed]= [malattia]\n",
    "            else:\n",
    "                validati_put_ulcer[put_seed].append(malattia)\n",
    "len(validati_put_ulcer), 'putative, su',  len(simb_49put_ulcer)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
